Search.setIndex({"docnames": ["README", "cnv/README", "gatk_data_preprocessing/README", "gatk_germline_snp_indel/README", "gatk_mitochondria_snp_indel/README", "gatk_somatic_snv_indel/README", "index", "meta_analysis_funnel_plot/README", "mutational_signatures/README", "sv/README", "tableone/README", "telomere_length/README", "year_one_screening_exam/README"], "filenames": ["README.md", "cnv/README.md", "gatk_data_preprocessing/README.md", "gatk_germline_snp_indel/README.md", "gatk_mitochondria_snp_indel/README.md", "gatk_somatic_snv_indel/README.md", "index.rst", "meta_analysis_funnel_plot/README.md", "mutational_signatures/README.md", "sv/README.md", "tableone/README.md", "telomere_length/README.md", "year_one_screening_exam/README.md"], "titles": ["phd-pipeline", "CNV calling pipeline", "Data pre-processing for variant discovery GATK4", "GATK4 germline short variant discovery (SNPs + Indels)", "GATK4 mitochondria SNPs and INDELs pipeline", "GATK4 call somatic snvs indels", "Welcome to phd-pipeline\u2019s documentation!", "Generate funnel plot for meta-analysis", "Mutational Signatures Pipelines", "Calling Structural Variations", "Create summary stats table using R tableone package", "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)", "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam &#64;usc"], "terms": {"cnv": 0, "epigenom": 0, "wide": 0, "associ": 0, "studi": [0, 7], "data": [0, 1, 4, 5], "pre": [0, 11], "process": [0, 5, 8, 11], "variant": [0, 11], "discoveri": [0, 11], "gatk4": 0, "germlin": [0, 4, 5], "short": 0, "snp": 0, "indel": 0, "mitochondri": [0, 4], "snv": 0, "call": 0, "somat": 0, "mediat": 0, "analysi": 0, "meta": 0, "funnel": 0, "plot": 0, "methyl": 0, "quantit": 0, "trait": 0, "loci": 0, "mutat": [0, 5], "signatur": 0, "sv": 0, "creat": [0, 5], "summari": [0, 9], "stat": 0, "tabl": [0, 5, 11], "telomer": 0, "length": 0, "estim": [0, 5], "year": 0, "one": [0, 5], "screen": 0, "exam": 0, "thi": [1, 2, 4, 5, 7, 8, 9], "i": [1, 4, 5, 8, 9], "copi": 1, "number": [1, 11], "variat": [1, 5], "from": [1, 3, 8, 9], "76": [1, 9], "match": [1, 2, 4, 5, 9], "tumor": [1, 2, 5, 9], "normal": [1, 2, 5, 9], "whole": [1, 2, 5, 9], "genom": [1, 2, 5, 9], "sequenc": [1, 2, 4, 5, 9], "wg": [1, 2, 3, 4, 5, 9, 11], "bam": [1, 5, 9], "38": [1, 9], "childhood": [1, 2, 4, 5, 9, 11], "acut": [1, 2, 4, 5, 9, 11], "lymphoblast": [1, 2, 4, 5, 9, 11], "leukemia": [1, 2, 4, 5, 9, 11], "case": [1, 2, 4, 5, 9, 11], "file": [1, 2, 8, 11], "have": [1, 5, 8], "been": [1, 8], "check": [1, 2, 3, 4, 5], "out": [1, 2, 3, 4, 5], "best": [1, 3, 4], "practic": [1, 2, 3, 4], "document": [1, 2, 3, 5, 11], "cnv_somatic_pair_workflow": 1, "wdl": [1, 2, 3, 4, 5], "how": [1, 5], "part": 1, "sensit": 1, "detect": [1, 5, 8], "ratio": 1, "alter": [1, 5], "allel": [1, 5], "segment": 1, "ii": 1, "legaci": 1, "tutori": [1, 2], "0": [1, 5], "extractintervalfrompon": 1, "sh": [1, 2, 4, 5, 9], "1": [1, 3, 9, 11], "9": [1, 4], "funcotateseg": 1, "workflow": [1, 3, 4, 5], "A": 1, "figur": 1, "gitc": 1, "2": [1, 11], "cnvkitaccess": 1, "3": [1, 9, 11], "iamp21cnvkitplot": 1, "cnvkittheta": 1, "cnvkitimporttheta": 1, "4": [1, 9, 11], "gisticcnvkitexportseg": 1, "gisticmergeseg": 1, "r": [1, 4, 8], "an": [2, 3, 4, 5, 9, 11], "hpc": [2, 3, 4, 5, 9], "50": [2, 5], "25": [2, 4, 5, 11], "were": [2, 5, 11], "illumina": [2, 11], "novaseq": [2, 11], "6000": [2, 11], "gatk": [2, 3, 5, 8, 11], "doc": [2, 4, 5], "more": [2, 3, 4, 5], "detail": [2, 3, 4, 5], "wa": [2, 5, 11], "written": [2, 5], "accord": [2, 3, 5], "follow": [2, 5], "offici": [2, 5], "version": 2, "http": [2, 3, 5, 7, 11], "github": [2, 3, 5, 11], "com": [2, 3, 5, 7, 11], "pipelin": [2, 3, 5, 9, 11], "blob": [2, 3], "master": [2, 3, 5], "task": [2, 3], "unmappedbamtoalignedbam": 2, "wholegenomegermlinesinglesampl": 2, "simpl": 2, "broadgsa": 2, "doc_arch": 2, "method": 2, "reference_implement": 2, "_pairedendsinglesamplewf_pipelin": 2, "md": 2, "sortandfixtag": 2, "other": [2, 5], "introduct": 2, "biowulf": 2, "nih": 2, "howto": 2, "run": [2, 3, 11], "bqsr": 2, "fastqc": 2, "pairedfastqtounmappedbam": 2, "validatesamfil": 2, "sortsam": 2, "bwa": 2, "sortalignedbam": 2, "validatealignedbam": 2, "checkifaltawarealign": 2, "mergebamalign": 2, "validatemergedbam": 2, "countread": 2, "validatesortedmarkedbam": 2, "baserecalibr": 2, "bqsrplot": 2, "rmd": [2, 8], "haplotypecal": 3, "gvcf": 3, "mode": [3, 5], "broad": 3, "prod": 3, "jointgenotypingwf": 3, "broadinstitut": [3, 5], "warp": 3, "jointgenotypingtask": 3, "dna_seq": 3, "joint_genotyp": 3, "jointgenotyp": 3, "larg": 4, "base": 4, "subsetbamtochrm": 4, "revertsam": 4, "aligntomt": 4, "aligntoshiftedmt": 4, "collectwgsmetr": 4, "runcollectwgsmetricsr": 4, "callmt": 4, "5": [4, 8, 9], "callshiftedmt": 4, "6": [4, 5], "liftoverandcombinevcf": 4, "7": [4, 11], "mergestat": [4, 5], "8": 4, "initialfilt": 4, "splitmultiallelicsandremovenonpasssit": 4, "getcontamin": 4, "filtercontamin": 4, "includ": [5, 11], "singl": 5, "nucleotid": 5, "sna": 5, "insert": 5, "delet": 5, "preprocess": 5, "tree": 5, "script": 5, "mutect2_wdl": 5, "explan": 5, "deprec": 5, "new": 5, "pdf": 5, "featur": 5, "improv": 5, "what": 5, "tradit": 5, "entail": 5, "faq": 5, "differ": [5, 8], "onli": 5, "versu": 5, "doe": 5, "do": 5, "each": [5, 11], "In": 5, "us": [5, 8, 9, 11], "popul": 5, "frequenc": 5, "resourc": 5, "prior": 5, "read": [5, 11], "fraction": 5, "\u00bd": 5, "ar": 5, "suggest": 5, "het": 5, "evid": 5, "miss": 5, "": 5, "same": 5, "model": 5, "addition": 5, "power": 5, "avail": 5, "exclud": 5, "rare": 5, "captur": 5, "individu": [5, 11], "specif": 5, "toward": 5, "being": 5, "The": 5, "simplifi": 5, "site": 5, "gnomad": 5, "retain": 5, "ftp": 5, "gsapubftp": 5, "anonym": 5, "org": 5, "bundl": 5, "panel": 5, "pon": 5, "ha": 5, "vital": 5, "role": 5, "fill": 5, "gap": 5, "between": 5, "catch": 5, "addit": 5, "nois": 5, "like": 5, "map": 5, "somewhat": 5, "random": 5, "systemat": 5, "interest": 5, "biostar": 5, "post": 5, "heterozygot": 5, "And": 5, "manifest": 5, "project": 5, "contain": 5, "artifactu": 5, "tool": 5, "take": [5, 11], "multipl": 5, "sampl": [5, 11], "callset": 5, "produc": 5, "collat": 5, "present": 5, "two": [5, 11], "default": 5, "set": 5, "min": 5, "count": 5, "argument": 5, "vcf": [5, 8], "common": 5, "level": 5, "max": 5, "probabl": 5, "threshold": [5, 11], "possibl": 5, "By": 5, "so": 5, "event": 5, "usual": 5, "correct": 5, "behavior": 5, "handl": 5, "probabilist": 5, "via": 5, "human": 5, "much": 5, "refin": 5, "than": 5, "ani": 5, "could": 5, "final": [5, 11], "tell": 5, "u": 5, "rate": 5, "cross": 5, "If": 5, "15": 5, "error": 5, "19": 5, "mean": 5, "everi": 5, "hundr": 5, "come": 5, "someon": 5, "els": 5, "percentag": 5, "effect": 5, "our": 5, "floor": 5, "low": 5, "potenti": 5, "below": 5, "af": 5, "we": [5, 11], "would": 5, "zero": 5, "judg": 5, "whether": 5, "thei": 5, "real": 5, "through": 5, "feed": 5, "can": 5, "properli": 5, "account": 5, "scatterinterv": 5, "mutect2scattergather0": 5, "mergevcf": 5, "mergebamout": 5, "orientationbia": 5, "createbwaimag": 5, "filteralignmentartifact": 5, "funcot": 5, "bcftoolfilt": 5, "fig2": 7, "fig3": 7, "www": 7, "liebertpub": 7, "doi": 7, "ab": 7, "10": 7, "1089": 7, "jicm": 7, "2022": 7, "0586": 7, "mutect2": 8, "algorithm": 8, "step1": 8, "py": 8, "step2": 8, "step3": 8, "step4": 8, "signalsummari": 8, "signalkataegi": 8, "step5": 8, "three": 9, "softwar": [9, 11], "premergesvpersamplesmoov": 9, "smooveunionsit": 9, "smoovegenotyp": 9, "smoovepasteannot": 9, "premergesvpersamplemantaconfigur": 9, "mantaworkflow": 9, "mantaclean": 9, "dellypersamplecal": 9, "dellyprefilt": 9, "dellygeno": 9, "dellypostfilt": 9, "dellybcf2vcf": 9, "survivor": 9, "anno": 9, "exampl": 9, "graph": 9, "appli": 11, "gener": 11, "compar": 11, "result": 11, "paper": 11, "content": 11, "quantifi": 11, "ten": 11, "repeat": 11, "tcaggg": 11, "tgaggg": 11, "ttgggg": 11, "ttcggg": 11, "tttggg": 11, "ataggg": 11, "catggg": 11, "ctaggg": 11, "gtaggg": 11, "taaggg": 11, "tl": 11, "wsa": 11, "kb": 11, "ttaggg": 11, "ccctaa": 11, "consid": 11, "step": 11, "aggreg": 11, "output": 11, "clean": 11, "calcul": 11, "need": 11, "weight": 11, "averag": 11, "all": 11, "group": 11, "within": 11, "zd1": 11, "issu": 11, "had": 11, "high": 11, "code": 12}, "objects": {}, "objtypes": {}, "objnames": {}, "titleterms": {"phd": [0, 6, 12], "pipelin": [0, 1, 4, 6, 8], "cnv": 1, "call": [1, 4, 5, 9], "gatk": [1, 4], "somat": [1, 5], "cnvkit": 1, "preprocess": [1, 2], "creat": [1, 9, 10], "theta": 1, "input": 1, "gistic": 1, "data": [2, 11], "pre": 2, "process": 2, "variant": [2, 3, 5, 9], "discoveri": [2, 3, 5], "gatk4": [2, 3, 4, 5], "The": 2, "main": 2, "step": [2, 4, 5], "workflow": [2, 11], "A": [2, 5, 11], "figur": [2, 5], "from": [2, 4, 5, 11], "gitc": [2, 5], "0": [2, 4], "qc": 2, "1": [2, 4, 5], "fastq": 2, "unmap": [2, 4], "bam": [2, 4, 11], "2": [2, 4, 5, 9], "map": 2, "3": [2, 4, 5], "merg": 2, "4": [2, 4, 5], "markdupl": 2, "5": [2, 5], "base": 2, "qualiti": 2, "score": 2, "recalibr": 2, "germlin": 3, "short": [3, 5], "snp": [3, 4], "indel": [3, 4, 5], "mitochondria": 4, "download": 4, "": [4, 6], "public": 4, "refer": 4, "databas": 4, "subset": 4, "whole": [4, 11], "genom": [4, 11], "just": 4, "read": 4, "remov": 4, "align": [4, 5], "inform": 4, "mark": 4, "duplic": 4, "collect": 4, "coverag": 4, "normal": [4, 11], "tumor": [4, 11], "estim": [4, 11, 12], "level": 4, "contamin": [4, 5], "snv": 5, "best": 5, "practic": 5, "scatter": 5, "interv": 5, "mutect2": 5, "parallel": 5, "gather": 5, "result": 5, "calcul": 5, "learn": 5, "orient": 5, "bia": 5, "filter": 5, "artifact": 5, "welcom": 6, "document": 6, "content": 6, "gener": 7, "funnel": 7, "plot": [7, 11], "meta": 7, "analysi": 7, "mutat": 8, "signatur": 8, "sigprofil": 8, "pmsignatur": 8, "deconstructsig": 8, "signal": 8, "websit": 8, "fitm": 8, "structur": 9, "variat": 9, "smoov": 9, "lumpi": 9, "manta": 9, "delli": 9, "find": 9, "were": 9, "least": 9, "caller": 9, "annot": 9, "venn": 9, "diagram": [9, 12], "histogram": 9, "html": 9, "file": 9, "summari": 10, "stat": 10, "tabl": 10, "us": 10, "r": 10, "tableon": 10, "packag": 10, "telomer": 11, "length": 11, "match": 11, "sequenc": 11, "background": 11, "telomerehunt": 11, "telseq": 11, "exampl": 11, "correl": 11, "sa": 12, "script": 12, "spider": 12, "i": 12, "prepar": 12, "epidemiologi": 12, "screen": 12, "exam": 12, "usc": 12, "appli": 12, "regress": 12, "predict": 12, "theori": 12}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinx.ext.viewcode": 1, "sphinx": 57}, "alltitles": {"Generate funnel plot for meta-analysis": [[7, "generate-funnel-plot-for-meta-analysis"]], "Mutational Signatures Pipelines": [[8, "mutational-signatures-pipelines"]], "Sigprofiler": [[8, "sigprofiler"]], "pmsignature": [[8, "pmsignature"]], "deconstructSigs": [[8, "deconstructsigs"]], "Signal website": [[8, "signal-website"]], "FitMS": [[8, "fitms"]], "Create summary stats table using R tableone package": [[10, "create-summary-stats-table-using-r-tableone-package"]], "Welcome to phd-pipeline\u2019s documentation!": [[6, "welcome-to-phd-pipeline-s-documentation"]], "Contents:": [[6, null]], "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)": [[11, "a-workflow-to-estimate-telomere-length-from-matched-tumor-normal-whole-genome-sequencing-data-bams"]], "Background": [[11, "background"]], "TelomereHunter": [[11, "telomerehunter"]], "TelSeq": [[11, "telseq"]], "Workflow": [[11, "workflow"]], "Example correlation plot": [[11, "example-correlation-plot"]], "Calling Structural Variations": [[9, "calling-structural-variations"]], "Smoove (lumpy)": [[9, "smoove-lumpy"]], "Manta": [[9, "manta"]], "Delly": [[9, "delly"]], "Finding variants that were called by at least 2 callers": [[9, "finding-variants-that-were-called-by-at-least-2-callers"]], "Annotate variants": [[9, "annotate-variants"]], "Create Venn diagram, histograms and html files": [[9, "create-venn-diagram-histograms-and-html-files"]], "CNV calling pipeline": [[1, "cnv-calling-pipeline"]], "GATK somatic CNV calling pipeline:": [[1, "gatk-somatic-cnv-calling-pipeline"]], "CNVkit pipeline": [[1, "cnvkit-pipeline"]], "Preprocessing to create Theta inputs": [[1, "preprocessing-to-create-theta-inputs"]], "Gistic pipeline": [[1, "gistic-pipeline"]], "GATK4 call somatic snvs indels": [[5, "gatk4-call-somatic-snvs-indels"]], "Best practices for somatic short variant discovery. A figure from GITC.": [[5, "best-practices-for-somatic-short-variant-discovery-a-figure-from-gitc"]], "Step 1: Scatter interval for Mutect2 parallel calling": [[5, "step-1-scatter-interval-for-mutect2-parallel-calling"]], "Step 2: Gather results from scattered intervals": [[5, "step-2-gather-results-from-scattered-intervals"]], "Step 3: Calculate contamination and learn orientation bias": [[5, "step-3-calculate-contamination-and-learn-orientation-bias"]], "Step 4: Filter alignment artifacts": [[5, "step-4-filter-alignment-artifacts"]], "Step 5: Filter variants": [[5, "step-5-filter-variants"]], "phd-pipeline": [[0, "phd-pipeline"]], "GATK4 germline short variant discovery (SNPs + Indels)": [[3, "gatk4-germline-short-variant-discovery-snps-indels"]], "GATK4 mitochondria SNPs and INDELs pipeline": [[4, "gatk4-mitochondria-snps-and-indels-pipeline"]], "Step 0: Download GATK\u2019s public reference databases": [[4, "step-0-download-gatk-s-public-reference-databases"]], "Step 1: Subset a whole genome bam to just Mitochondria reads and remove alignment information": [[4, "step-1-subset-a-whole-genome-bam-to-just-mitochondria-reads-and-remove-alignment-information"]], "Step 2: Align unmapped bam, mark duplicates, collect coverage information, and call SNPs and INDELs from normal bams": [[4, "step-2-align-unmapped-bam-mark-duplicates-collect-coverage-information-and-call-snps-and-indels-from-normal-bams"]], "Step 3: Call SNPs and INDELs from tumor bams": [[4, "step-3-call-snps-and-indels-from-tumor-bams"]], "Step 4: Estimate levels of contamination in mitochondria": [[4, "step-4-estimate-levels-of-contamination-in-mitochondria"]], "Data pre-processing for variant discovery GATK4": [[2, "data-pre-processing-for-variant-discovery-gatk4"]], "The main steps in the preprocessing workflow. A figure from GITC.": [[2, "the-main-steps-in-the-preprocessing-workflow-a-figure-from-gitc"]], "Step 0: QC": [[2, "step-0-qc"]], "Step 1: Fastqs to Unmapped BAMs": [[2, "step-1-fastqs-to-unmapped-bams"]], "Step 2: Unmapped BAMs to Mapped BAMs": [[2, "step-2-unmapped-bams-to-mapped-bams"]], "Step 3: Merge Unmapped BAMs and Mapped BAMs": [[2, "step-3-merge-unmapped-bams-and-mapped-bams"]], "Step 4: MarkDuplicates": [[2, "step-4-markduplicates"]], "Step 5: Base Quality Score Recalibration": [[2, "step-5-base-quality-score-recalibration"]], "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam @usc": [[12, "sas-scripts-and-spider-diagram-that-i-prepared-for-the-epidemiology-phd-screening-exam-usc"]], "apply exam - regression estimation and prediction": [[12, "apply-exam-regression-estimation-and-prediction"]], "theory - epidemiology": [[12, "theory-epidemiology"]]}, "indexentries": {}})