Search.setIndex({"docnames": ["cnv/README", "data_visualization/chord_diagram/README", "data_visualization/global_ancestry_pca/README", "data_visualization/heatmap/README", "data_visualization/meta_analysis_forest_plot/README", "data_visualization/meta_analysis_funnel_plot/README", "data_visualization/regression_coefficient_ci/README", "data_visualization/sample_flow_chart/README", "data_visualization/snp_cpg_association/README", "data_visualization/ttest_boxplot/README", "epigenetic_clock/README", "ewas/README", "gatk_data_preprocessing/README", "gatk_germline_snp_indel/README", "gatk_mitochondria_snp_indel/README", "gatk_somatic_snv_indel/README", "gwas/README", "index", "mediation_analysis/README", "mqtl/README", "mutational_signatures/README", "sv/README", "tableone/README", "telomere_length/README", "year_one_screening_exam/README"], "filenames": ["cnv/README.md", "data_visualization/chord_diagram/README.md", "data_visualization/global_ancestry_pca/README.md", "data_visualization/heatmap/README.md", "data_visualization/meta_analysis_forest_plot/README.md", "data_visualization/meta_analysis_funnel_plot/README.md", "data_visualization/regression_coefficient_ci/README.md", "data_visualization/sample_flow_chart/README.md", "data_visualization/snp_cpg_association/README.md", "data_visualization/ttest_boxplot/README.md", "epigenetic_clock/README.md", "ewas/README.md", "gatk_data_preprocessing/README.md", "gatk_germline_snp_indel/README.md", "gatk_mitochondria_snp_indel/README.md", "gatk_somatic_snv_indel/README.md", "gwas/README.md", "index.rst", "mediation_analysis/README.md", "mqtl/README.md", "mutational_signatures/README.md", "sv/README.md", "tableone/README.md", "telomere_length/README.md", "year_one_screening_exam/README.md"], "titles": ["CNV calling pipeline", "Chord diagram", "Global ancestry pca", "Correlation heatmap", "Meta analysis forest plot", "Generate funnel plot for meta-analysis", "Regression coefficients and CIs", "Sample flow chart", "SNP CpG association", "t-test boxplot", "Eigenetic clock", "Epigenome-wide association study", "Data pre-processing for variant discovery GATK4", "GATK4 germline short variant discovery (SNPs + Indels)", "GATK4 mitochondria SNPs and INDELs pipeline", "GATK4 call somatic snvs indels", "Genome-wide association study", "phd-pipeline", "Causal mediation analysis", "Methylation quantitative trait loci analysis", "Mutational Signatures Pipelines", "Calling Structural Variations", "Create summary stats table using R tableone package", "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)", "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam &#64;usc"], "terms": {"thi": [0, 1, 3, 4, 5, 6, 7, 9, 10, 12, 14, 15, 16, 18, 19, 20, 21], "i": [0, 1, 3, 4, 6, 8, 9, 10, 14, 15, 16, 17, 18, 19, 20, 21], "copi": 0, "number": [0, 1, 4, 11, 19, 23], "variat": [0, 15], "from": [0, 8, 9, 10, 11, 13, 16, 20, 21], "76": [0, 21], "match": [0, 12, 14, 15, 21], "tumor": [0, 1, 12, 15, 21], "normal": [0, 12, 15, 18, 21], "whole": [0, 12, 15, 21], "genom": [0, 7, 8, 12, 15, 17, 21], "sequenc": [0, 12, 14, 15, 21], "wg": [0, 12, 13, 14, 15, 21, 23], "bam": [0, 15, 21], "38": [0, 21], "childhood": [0, 12, 14, 15, 19, 21, 23], "acut": [0, 12, 14, 15, 21, 23], "lymphoblast": [0, 12, 14, 15, 21, 23], "leukemia": [0, 12, 14, 15, 21, 23], "case": [0, 1, 4, 7, 8, 9, 11, 12, 14, 15, 16, 21, 23], "file": [0, 12, 20, 23], "have": [0, 1, 7, 15, 20], "been": [0, 20], "data": [0, 7, 8, 10, 14, 15, 16], "check": [0, 12, 13, 14, 15], "out": [0, 12, 13, 14, 15], "best": [0, 13, 14], "practic": [0, 12, 13, 14], "document": [0, 12, 13, 15, 23], "cnv_somatic_pair_workflow": 0, "wdl": [0, 12, 13, 14, 15], "how": [0, 15], "part": 0, "sensit": 0, "detect": [0, 15, 20], "ratio": [0, 4], "alter": [0, 15], "allel": [0, 8, 15], "segment": 0, "ii": [0, 18], "legaci": 0, "tutori": [0, 12], "0": [0, 4, 11, 15, 16, 19], "extractintervalfrompon": 0, "sh": [0, 12, 14, 15, 16, 21], "1": [0, 4, 11, 13, 16, 19, 21, 23], "9": [0, 14], "funcotateseg": 0, "workflow": [0, 10, 11, 13, 14, 15, 16, 17, 18, 19], "A": [0, 8, 10, 11], "figur": 0, "gitc": 0, "2": [0, 8, 16, 19, 23], "cnvkitaccess": 0, "3": [0, 16, 21, 23], "iamp21cnvkitplot": 0, "cnvkittheta": 0, "cnvkitimporttheta": 0, "4": [0, 4, 11, 21, 23], "gisticcnvkitexportseg": 0, "gisticmergeseg": 0, "r": [0, 1, 8, 10, 11, 14, 18, 19, 20], "script": [1, 3, 4, 6, 9, 11, 15, 18, 19, 22], "gener": [1, 10, 11, 22, 23], "show": [1, 3, 4, 6, 8, 17, 18], "gene": [1, 4, 7, 8, 11], "delet": [1, 4, 7, 15], "frequenc": [1, 4, 7, 8, 15], "eight": 1, "commonli": 1, "region": [1, 8], "cdkn2a": 1, "etv6": 1, "ikzf1": [1, 8], "pax5": 1, "btg1": 1, "ebf1": 1, "rb1": 1, "within": [1, 23], "pseudoautosom": 1, "par1": 1, "sex": [1, 11, 18], "chromosom": [1, 8, 11], "crlf2": 1, "csf2ra": 1, "il3ra": 1, "thei": [1, 8, 15], "were": [1, 3, 4, 7, 8, 9, 10, 11, 12, 15, 16, 18, 19, 23], "assai": 1, "dna": [1, 3, 4, 6, 7, 8, 9, 10, 11, 19], "us": [1, 3, 9, 10, 11, 15, 16, 18, 19, 20, 21, 23], "multiplex": 1, "ligat": 1, "depend": [1, 19], "probe": 1, "amplif": 1, "mlpa": [1, 7], "The": [1, 4, 8, 9, 10, 15, 16, 18, 19], "proport": [1, 4], "least": 1, "one": [1, 15, 17, 22], "zero": [1, 15], "epic": [1, 3, 4, 7, 9, 10], "arrai": [1, 3, 7, 9, 10], "n": [1, 3, 7, 9], "284": [1, 3, 7, 9], "dataset": [1, 3, 4, 9, 16, 18, 19], "width": 1, "end": 1, "each": [1, 4, 11, 15, 19, 23], "link": [1, 3, 4, 6, 7, 8, 9, 10, 11, 18, 19, 22], "between": [1, 4, 7, 8, 9, 15, 16, 19], "two": [1, 4, 9, 10, 11, 15, 18, 23], "both": 1, "creat": [1, 3, 4, 6, 15, 17], "packag": [1, 11, 18, 19], "circliz": 1, "studi": [1, 3, 4, 5, 6, 7, 8, 9, 10, 18, 19, 22], "public": [1, 3, 4, 6, 7, 8, 9, 10, 11, 18, 19], "output": [1, 3, 4, 6, 7, 9, 23], "visual": [2, 8], "smartpca": [2, 16], "result": [2, 4, 6, 11, 18, 19, 23], "smartpca_plot": 2, "rmd": [2, 8, 10, 11, 12, 20], "admixtur": [2, 16], "admixture_plot": 2, "pearson": 3, "coeffici": [3, 17], "self": [3, 6], "report": [3, 6], "smoke": [3, 4, 6, 7, 9], "variabl": [3, 4], "It": 3, "matric": 3, "parent": [3, 6], "tobacco": [3, 4, 6, 7, 9], "exposur": [3, 4, 6, 7], "methyl": [3, 4, 6, 7, 8, 9, 10, 11, 17, 18], "ahrr": [3, 4, 6, 7], "cpg": [3, 4, 6, 7, 9, 10, 11, 17, 19], "cg05575921": [3, 4, 6, 7], "polyepigenet": [3, 4, 6, 9], "score": [3, 4, 6, 9, 16], "p": [3, 9], "valu": [3, 4, 9, 10, 11], "calcul": [3, 4, 9, 10, 23], "spearman": 3, "": [3, 15], "rank": 3, "test": [3, 17, 19], "statist": [3, 18], "signific": [3, 18], "mark": 3, "asterisk": 3, "beta": [3, 9, 10, 11], "wa": [3, 4, 10, 12, 15, 16, 18, 19, 23], "multipli": 3, "10": [3, 5, 11], "associ": [4, 6, 7, 9, 19], "epigenet": [4, 10, 17], "biomark": 4, "prenat": [4, 7], "b": [4, 7], "all": [4, 7, 8, 11, 16, 18, 19, 23], "panel": [4, 8, 9, 15], "includ": [4, 7, 10, 11, 15, 18, 23], "poisson": 4, "regress": [4, 11, 16, 17, 19], "top": [4, 8], "middl": [4, 8], "exclud": [4, 15], "bottom": 4, "mean": [4, 15], "rm": 4, "everi": [4, 15, 16], "decreas": 4, "unit": 4, "increas": 4, "model": [4, 11, 15, 18, 19], "adjust": [4, 11, 19], "cell": [4, 11, 16, 18], "type": [4, 11, 16, 18], "heterogen": [4, 11, 18], "genet": [4, 11, 16], "ancestri": [4, 11, 16, 17, 18], "addition": [4, 15], "qtl": 4, "snp": [4, 16, 17, 19], "genotyp": [4, 8, 16, 18, 19], "rs148405299": 4, "450k": [4, 7, 9, 10], "rs77111113": 4, "center": 4, "squar": 4, "horizont": [4, 11], "bar": 4, "through": [4, 15], "indic": [4, 8], "point": [4, 8], "estim": [4, 10, 11, 15, 16, 17, 18], "95": 4, "ci": [4, 17], "individu": [4, 15, 23], "set": [4, 11, 15], "area": 4, "rel": 4, "weight": [4, 10, 23], "vertic": 4, "apic": 4, "diamond": 4, "summari": [4, 11, 17, 21], "reciproc": 4, "sampl": [4, 9, 11, 15, 17, 22, 23], "varianc": [4, 18], "srm": 4, "ar": [4, 7, 8, 9, 10, 11, 15], "summar": [4, 18], "right": [4, 7, 8], "fig2": 5, "fig3": 5, "http": [5, 12, 13, 15, 23], "www": 5, "liebertpub": 5, "com": [5, 12, 13, 15, 23], "doi": 5, "ab": 5, "1089": 5, "jicm": 5, "2022": 5, "0586": 5, "plot": [6, 10, 11, 17], "multipl": [6, 15, 19], "linear": [6, 16, 19], "flowchart": 7, "left": [7, 8], "box": 7, "previou": 7, "analysi": [7, 11, 16, 17], "earli": 7, "life": 7, "559": 7, "which": [7, 18], "361": 7, "analyz": [7, 16, 19], "onli": [7, 15], "interview": 7, "198": [7, 9], "had": [7, 23], "illumina": [7, 10, 12, 23], "wide": [7, 17], "avail": [7, 10, 15], "thu": [7, 10], "analys": [7, 11, 18], "previous": 7, "now": 7, "requenc": 7, "178": 7, "overlap": [7, 10, 11, 17], "In": [7, 15], "total": [7, 18], "482": [7, 9], "our": [7, 15], "eleven": 7, "106": 7, "ldproxi": 8, "proxi": 8, "variant": [8, 16, 17, 23], "rs6421315": 8, "european": 8, "eur": 8, "popul": [8, 15, 16], "v": [8, 16], "rs78396808": 8, "admix": 8, "american": 8, "amr": 8, "gviz": 8, "locat": 8, "dmp": [8, 18, 19], "cg01139861": 8, "cg10551353": 8, "incorpor": 8, "annot": 8, "queri": 8, "ucsc": 8, "browser": 8, "via": [8, 15], "track": 8, "ideogram": 8, "7": [8, 14, 23], "black": 8, "rectangl": 8, "where": 8, "second": 8, "axi": [8, 11], "start": 8, "posit": [8, 9, 11], "50342500": 8, "50472798": 8, "refer": [8, 16], "build": 8, "hg19": 8, "third": 8, "transcript": 8, "fourth": 8, "three": [8, 10, 16, 18, 19, 21], "island": 8, "chr7": 8, "50342895": 8, "50": [8, 12, 15], "343": 8, "456": 8, "46": 8, "50343757": 8, "344": 8, "519": 8, "80": 8, "50467566": 8, "50468400": 8, "79": 8, "last": 8, "50343456": 8, "promot": 8, "an": [8, 12, 13, 14, 15, 18, 19, 21, 23], "intron": 8, "116": 8, "kb": [8, 23], "downstream": 8, "5": [8, 14, 16, 20, 21], "utr": 8, "tss1500": 8, "sever": 8, "14": 8, "snp_cpg_associ": 8, "relationship": 8, "level": [8, 10, 15, 18], "repres": [8, 9, 11], "median": 8, "risk": [8, 16, 19], "control": [8, 11, 16], "overal": [8, 9, 11], "latino": 8, "non": [8, 18], "white": 8, "independ": [9, 16], "compar": [9, 23], "share": 9, "plu": 9, "minu": 9, "sign": [9, 16], "label": [9, 11], "whether": [9, 15], "neg": 9, "matern": 9, "obtain": 9, "five": 10, "newborn": 10, "down": 10, "syndrom": 10, "methylationep": 10, "These": 10, "ag": 10, "dnamskinbloodclock": 10, "pan": 10, "tissu": 10, "dnamag": 10, "gestat": 10, "haftorn": 10, "knight": 10, "bohlin": 10, "base": [10, 14, 16, 18], "accord": [10, 12, 13, 15], "method": [10, 12], "horvath": 10, "2013": 10, "et": [10, 16], "al": [10, 16], "2018": 10, "skin": 10, "blood": [10, 16], "briefli": 10, "calibr": 10, "version": [10, 12], "combin": 10, "353": 10, "subsequ": [10, 19], "convert": 10, "back": 10, "similar": 10, "wai": 10, "391": 10, "2016": 10, "2021": 10, "compris": 10, "176": 10, "compos": 10, "148": 10, "96": 10, "respect": [10, 11], "six": 10, "8": [10, 14], "absent": 10, "replac": 10, "averag": [10, 23], "142": 10, "88": 10, "imput": [10, 16], "knn": 10, "function": [10, 11, 16, 18, 19], "full": 10, "list": 10, "can": [10, 15], "found": 10, "here": 10, "epigenetic_clock": 10, "heatmap_cpg": 10, "heatmap": [10, 17], "ga_clock_corr": 10, "correl": [10, 17], "run": [11, 12, 13, 23], "ewa": 11, "identifi": [11, 16, 19], "differenti": 11, "autosom": [11, 16], "we": [11, 15, 16, 18, 19, 23], "fit": [11, 16, 18, 19], "logist": [11, 16], "predict": [11, 16, 19], "statu": [11, 16, 18], "remain": 11, "batch": [11, 18], "effect": [11, 15, 18, 19], "first": [11, 18, 19], "princip": [11, 16], "compon": [11, 16], "deriv": 11, "refactor": 11, "58": 11, "epistructur": 11, "fix": [11, 18, 19], "meta": [11, 17, 18, 19], "hree": 11, "differ": [11, 15, 20], "metafor": [11, 19], "bidirect": [11, 17], "manhattan": [11, 17], "qq": [11, 17], "cmplot": 11, "line": 11, "bonferroni": [11, 19], "threshold": [11, 15, 19, 23], "025": [11, 19], "divid": 11, "solid": 11, "lenient": 11, "dash": 11, "y": 11, "log10phyp": 11, "hypermethyl": 11, "higher": 11, "versu": [11, 15], "log10phypo": 11, "hypomethyl": 11, "lower": 11, "later": 11, "causal": [11, 17], "mediat": [11, 17], "name": 11, "hpc": [12, 13, 14, 15, 21], "25": [12, 14, 15, 23], "novaseq": [12, 23], "6000": [12, 23], "gatk": [12, 13, 15, 20, 23], "doc": [12, 14, 15], "more": [12, 13, 14, 15], "detail": [12, 13, 14, 15], "written": [12, 15], "follow": [12, 15], "offici": [12, 15], "github": [12, 13, 15, 23], "pipelin": [12, 13, 15, 21, 23], "blob": [12, 13], "master": [12, 13, 15], "task": [12, 13], "unmappedbamtoalignedbam": 12, "wholegenomegermlinesinglesampl": 12, "simpl": 12, "broadgsa": 12, "doc_arch": 12, "reference_implement": 12, "_pairedendsinglesamplewf_pipelin": 12, "md": 12, "sortandfixtag": 12, "other": [12, 15], "introduct": 12, "biowulf": 12, "nih": 12, "howto": 12, "bqsr": 12, "fastqc": 12, "pairedfastqtounmappedbam": 12, "validatesamfil": 12, "sortsam": 12, "bwa": 12, "sortalignedbam": 12, "validatealignedbam": 12, "checkifaltawarealign": 12, "mergebamalign": 12, "validatemergedbam": 12, "countread": 12, "validatesortedmarkedbam": 12, "baserecalibr": 12, "bqsrplot": 12, "haplotypecal": 13, "gvcf": 13, "mode": [13, 15], "broad": 13, "prod": 13, "jointgenotypingwf": 13, "broadinstitut": [13, 15], "warp": 13, "jointgenotypingtask": 13, "dna_seq": 13, "joint_genotyp": 13, "jointgenotyp": 13, "mitochondri": [14, 17], "larg": 14, "subsetbamtochrm": 14, "revertsam": 14, "aligntomt": 14, "aligntoshiftedmt": 14, "collectwgsmetr": 14, "runcollectwgsmetricsr": 14, "callmt": 14, "germlin": [14, 15, 17], "callshiftedmt": 14, "6": [14, 15], "liftoverandcombinevcf": 14, "mergestat": [14, 15], "initialfilt": 14, "splitmultiallelicsandremovenonpasssit": 14, "getcontamin": 14, "filtercontamin": 14, "mutat": [15, 17], "singl": 15, "nucleotid": 15, "sna": 15, "insert": 15, "preprocess": 15, "tree": 15, "mutect2_wdl": 15, "explan": 15, "deprec": 15, "new": 15, "pdf": 15, "featur": 15, "improv": 15, "what": 15, "tradit": 15, "entail": 15, "faq": 15, "doe": 15, "do": 15, "resourc": 15, "prior": 15, "read": [15, 23], "fraction": 15, "\u00bd": 15, "suggest": 15, "het": 15, "evid": 15, "miss": 15, "same": [15, 18], "power": 15, "rare": 15, "captur": 15, "specif": 15, "toward": 15, "being": 15, "simplifi": 15, "site": [15, 19], "gnomad": 15, "retain": 15, "ftp": 15, "gsapubftp": 15, "anonym": 15, "org": 15, "bundl": 15, "pon": 15, "ha": 15, "vital": 15, "role": 15, "fill": 15, "gap": 15, "catch": 15, "addit": [15, 19], "nois": 15, "like": 15, "map": 15, "somewhat": 15, "random": [15, 16], "systemat": 15, "process": [15, 17, 20, 23], "interest": 15, "biostar": 15, "post": [15, 16], "heterozygot": 15, "And": 15, "manifest": 15, "project": [15, 17], "contain": 15, "artifactu": 15, "tool": 15, "take": [15, 23], "callset": 15, "produc": 15, "collat": 15, "present": 15, "default": 15, "min": 15, "count": 15, "argument": 15, "vcf": [15, 20], "common": 15, "max": 15, "probabl": 15, "possibl": 15, "By": 15, "so": 15, "event": 15, "usual": 15, "correct": [15, 19], "behavior": 15, "handl": 15, "probabilist": 15, "human": 15, "much": 15, "refin": 15, "than": 15, "ani": 15, "could": 15, "final": [15, 23], "tabl": [15, 17, 23], "tell": 15, "u": 15, "rate": 15, "cross": 15, "If": 15, "15": 15, "error": 15, "19": 15, "hundr": 15, "come": 15, "someon": 15, "els": 15, "percentag": 15, "floor": 15, "low": 15, "potenti": 15, "below": 15, "af": 15, "would": 15, "judg": 15, "real": 15, "feed": 15, "properli": 15, "account": 15, "scatterinterv": 15, "mutect2scattergather0": 15, "mergevcf": 15, "mergebamout": 15, "orientationbia": 15, "createbwaimag": 15, "filteralignmentartifact": 15, "funcot": 15, "bcftoolfilt": 15, "perform": [16, 18], "gwa": 16, "affy6": 16, "500k": 16, "pre_imputation_qc": 16, "pre": [16, 17, 23], "qualiti": 16, "anderson": 16, "mare": 16, "pca_script": 16, "stratif": 16, "next": 16, "global": [16, 17], "v1": 16, "pca": [16, 17], "program": 16, "eigensoft": 16, "v6": 16, "conduct": [16, 18, 19], "michigan": 16, "server": 16, "post_imput": 16, "maf": 16, "r2": 16, "filter": 16, "pc": 16, "comput": [16, 18], "dosag": 16, "inform": [16, 22], "subset": 16, "linkag": 16, "equilibrium": 16, "plink": 16, "v2": 16, "1kgp": 16, "phase": 16, "gwas_run": 16, "gwas_ordinal_logist": 16, "phenotyp": 16, "central": 16, "nervou": 16, "system": 16, "involv": 16, "classifi": 16, "cns1": 16, "blast": 16, "csf": 16, "cns2": 16, "wbc": 16, "mul": 16, "cns3": 16, "cn": 16, "To": 16, "firth": 16, "binomi": 16, "ordin": 16, "ordinalgwa": 16, "jl": 16, "pr": 16, "polygenet": 16, "trait": [16, 17], "some": 17, "wrote": 17, "my": 17, "research": 17, "discoveri": [17, 23], "gatk4": 17, "short": 17, "indel": 17, "call": 17, "somat": 17, "snv": 17, "sv": 17, "cnv": 17, "signatur": 17, "telomer": 17, "length": 17, "eigenet": 17, "clock": 17, "epigenom": 17, "quantit": 17, "loci": 17, "stat": 17, "chord": 17, "diagram": [17, 18], "funnel": 17, "forest": 17, "flow": 17, "chart": 17, "t": 17, "boxplot": 17, "year": 17, "screen": 17, "exam": 17, "mqtl": [18, 19], "pair": 18, "specifi": 18, "distribut": 18, "after": 18, "condit": 18, "covari": 18, "outcom": 18, "given": 18, "separ": [18, 19], "paramet": 18, "main": 18, "input": 18, "acm": 18, "ad": 18, "simul": 18, "quasi": 18, "bayesian": 18, "mont": 18, "carlo": 18, "approxim": 18, "1000": 18, "time": 18, "altern": 18, "approach": 18, "parametr": 18, "bootstrap": 18, "also": 18, "appli": [18, 23], "valid": 18, "across": [18, 19], "path": 18, "confound": 18, "matrix": 19, "eqtl": 19, "code": [19, 24], "stringent": 19, "fdr": 19, "05": 19, "mutect2": 20, "algorithm": 20, "step1": 20, "py": 20, "step2": 20, "step3": 20, "step4": 20, "signalsummari": 20, "signalkataegi": 20, "step5": 20, "softwar": [21, 23], "premergesvpersamplesmoov": 21, "smooveunionsit": 21, "smoovegenotyp": 21, "smoovepasteannot": 21, "premergesvpersamplemantaconfigur": 21, "mantaworkflow": 21, "mantaclean": 21, "dellypersamplecal": 21, "dellyprefilt": 21, "dellygeno": 21, "dellypostfilt": 21, "dellybcf2vcf": 21, "survivor": 21, "anno": 21, "exampl": 21, "graph": 21, "paper": 23, "content": 23, "quantifi": 23, "ten": 23, "repeat": 23, "tcaggg": 23, "tgaggg": 23, "ttgggg": 23, "ttcggg": 23, "tttggg": 23, "ataggg": 23, "catggg": 23, "ctaggg": 23, "gtaggg": 23, "taaggg": 23, "tl": 23, "wsa": 23, "ttaggg": 23, "ccctaa": 23, "consid": 23, "step": 23, "aggreg": 23, "clean": 23, "need": 23, "group": 23, "zd1": 23, "issu": 23, "high": 23}, "objects": {}, "objtypes": {}, "objnames": {}, "titleterms": {"cnv": 0, "call": [0, 14, 15, 21], "pipelin": [0, 14, 17, 20], "gatk": [0, 14], "somat": [0, 15], "cnvkit": 0, "preprocess": [0, 12], "creat": [0, 21, 22], "theta": 0, "input": 0, "gistic": 0, "chord": 1, "diagram": [1, 21, 24], "global": 2, "ancestri": 2, "pca": 2, "correl": [3, 23], "heatmap": 3, "meta": [4, 5], "analysi": [4, 5, 18, 19], "forest": 4, "plot": [4, 5, 23], "gener": 5, "funnel": 5, "regress": [6, 24], "coeffici": 6, "ci": 6, "sampl": 7, "flow": 7, "chart": 7, "snp": [8, 13, 14], "cpg": 8, "associ": [8, 11, 16, 17], "t": 9, "test": 9, "boxplot": 9, "eigenet": 10, "clock": 10, "epigenom": 11, "wide": [11, 16], "studi": [11, 16, 17], "data": [12, 17, 23], "pre": 12, "process": 12, "variant": [12, 13, 15, 21], "discoveri": [12, 13, 15], "gatk4": [12, 13, 14, 15], "The": 12, "main": 12, "step": [12, 14, 15], "workflow": [12, 23], "A": [12, 15, 23], "figur": [12, 15], "from": [12, 14, 15, 23], "gitc": [12, 15], "0": [12, 14], "qc": 12, "1": [12, 14, 15], "fastq": 12, "unmap": [12, 14], "bam": [12, 14, 23], "2": [12, 14, 15, 21], "map": 12, "3": [12, 14, 15], "merg": 12, "4": [12, 14, 15], "markdupl": 12, "5": [12, 15], "base": 12, "qualiti": 12, "score": 12, "recalibr": 12, "germlin": 13, "short": [13, 15], "indel": [13, 14, 15], "mitochondria": 14, "download": 14, "": 14, "public": 14, "refer": 14, "databas": 14, "subset": 14, "whole": [14, 23], "genom": [14, 16, 23], "just": 14, "read": 14, "remov": 14, "align": [14, 15], "inform": 14, "mark": 14, "duplic": 14, "collect": 14, "coverag": 14, "normal": [14, 23], "tumor": [14, 23], "estim": [14, 23, 24], "level": 14, "contamin": [14, 15], "snv": 15, "best": 15, "practic": 15, "scatter": 15, "interv": 15, "mutect2": 15, "parallel": 15, "gather": 15, "result": 15, "calcul": 15, "learn": 15, "orient": 15, "bia": 15, "filter": 15, "artifact": 15, "phd": [17, 24], "bioinformat": 17, "visual": 17, "other": 17, "causal": 18, "mediat": 18, "methyl": 19, "quantit": 19, "trait": 19, "loci": 19, "mutat": 20, "signatur": 20, "sigprofil": 20, "pmsignatur": 20, "deconstructsig": 20, "signal": 20, "websit": 20, "fitm": 20, "structur": 21, "variat": 21, "smoov": 21, "lumpi": 21, "manta": 21, "delli": 21, "find": 21, "were": 21, "least": 21, "caller": 21, "annot": 21, "venn": 21, "histogram": 21, "html": 21, "file": 21, "summari": 22, "stat": 22, "tabl": 22, "us": 22, "r": 22, "tableon": 22, "packag": 22, "telomer": 23, "length": 23, "match": 23, "sequenc": 23, "background": 23, "telomerehunt": 23, "telseq": 23, "exampl": 23, "sa": 24, "script": 24, "spider": 24, "i": 24, "prepar": 24, "epidemiologi": 24, "screen": 24, "exam": 24, "usc": 24, "appli": 24, "predict": 24, "theori": 24}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinx.ext.viewcode": 1, "sphinx": 57}, "alltitles": {"Chord diagram": [[1, "chord-diagram"]], "SNP CpG association": [[8, "snp-cpg-association"]], "Sample flow chart": [[7, "sample-flow-chart"]], "Generate funnel plot for meta-analysis": [[5, "generate-funnel-plot-for-meta-analysis"]], "Meta analysis forest plot": [[4, "meta-analysis-forest-plot"]], "Correlation heatmap": [[3, "correlation-heatmap"]], "CNV calling pipeline": [[0, "cnv-calling-pipeline"]], "GATK somatic CNV calling pipeline:": [[0, "gatk-somatic-cnv-calling-pipeline"]], "CNVkit pipeline": [[0, "cnvkit-pipeline"]], "Preprocessing to create Theta inputs": [[0, "preprocessing-to-create-theta-inputs"]], "Gistic pipeline": [[0, "gistic-pipeline"]], "Eigenetic clock": [[10, "eigenetic-clock"]], "Global ancestry pca": [[2, "global-ancestry-pca"]], "Regression coefficients and CIs": [[6, "regression-coefficients-and-cis"]], "t-test boxplot": [[9, "t-test-boxplot"]], "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam @usc": [[24, "sas-scripts-and-spider-diagram-that-i-prepared-for-the-epidemiology-phd-screening-exam-usc"]], "apply exam - regression estimation and prediction": [[24, "apply-exam-regression-estimation-and-prediction"]], "theory - epidemiology": [[24, "theory-epidemiology"]], "Create summary stats table using R tableone package": [[22, "create-summary-stats-table-using-r-tableone-package"]], "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)": [[23, "a-workflow-to-estimate-telomere-length-from-matched-tumor-normal-whole-genome-sequencing-data-bams"]], "Background": [[23, "background"]], "TelomereHunter": [[23, "telomerehunter"]], "TelSeq": [[23, "telseq"]], "Workflow": [[23, "workflow"]], "Example correlation plot": [[23, "example-correlation-plot"]], "Genome-wide association study": [[16, "genome-wide-association-study"]], "GATK4 call somatic snvs indels": [[15, "gatk4-call-somatic-snvs-indels"]], "Best practices for somatic short variant discovery. A figure from GITC.": [[15, "best-practices-for-somatic-short-variant-discovery-a-figure-from-gitc"]], "Step 1: Scatter interval for Mutect2 parallel calling": [[15, "step-1-scatter-interval-for-mutect2-parallel-calling"]], "Step 2: Gather results from scattered intervals": [[15, "step-2-gather-results-from-scattered-intervals"]], "Step 3: Calculate contamination and learn orientation bias": [[15, "step-3-calculate-contamination-and-learn-orientation-bias"]], "Step 4: Filter alignment artifacts": [[15, "step-4-filter-alignment-artifacts"]], "Step 5: Filter variants": [[15, "step-5-filter-variants"]], "Methylation quantitative trait loci analysis": [[19, "methylation-quantitative-trait-loci-analysis"]], "Calling Structural Variations": [[21, "calling-structural-variations"]], "Smoove (lumpy)": [[21, "smoove-lumpy"]], "Manta": [[21, "manta"]], "Delly": [[21, "delly"]], "Finding variants that were called by at least 2 callers": [[21, "finding-variants-that-were-called-by-at-least-2-callers"]], "Annotate variants": [[21, "annotate-variants"]], "Create Venn diagram, histograms and html files": [[21, "create-venn-diagram-histograms-and-html-files"]], "phd-pipeline": [[17, "phd-pipeline"]], "Bioinformatics pipelines": [[17, "bioinformatics-pipelines"]], "Association studies": [[17, "association-studies"]], "Data visualizations": [[17, "data-visualizations"]], "Other": [[17, "other"]], "Data pre-processing for variant discovery GATK4": [[12, "data-pre-processing-for-variant-discovery-gatk4"]], "The main steps in the preprocessing workflow. A figure from GITC.": [[12, "the-main-steps-in-the-preprocessing-workflow-a-figure-from-gitc"]], "Step 0: QC": [[12, "step-0-qc"]], "Step 1: Fastqs to Unmapped BAMs": [[12, "step-1-fastqs-to-unmapped-bams"]], "Step 2: Unmapped BAMs to Mapped BAMs": [[12, "step-2-unmapped-bams-to-mapped-bams"]], "Step 3: Merge Unmapped BAMs and Mapped BAMs": [[12, "step-3-merge-unmapped-bams-and-mapped-bams"]], "Step 4: MarkDuplicates": [[12, "step-4-markduplicates"]], "Step 5: Base Quality Score Recalibration": [[12, "step-5-base-quality-score-recalibration"]], "Mutational Signatures Pipelines": [[20, "mutational-signatures-pipelines"]], "Sigprofiler": [[20, "sigprofiler"]], "pmsignature": [[20, "pmsignature"]], "deconstructSigs": [[20, "deconstructsigs"]], "Signal website": [[20, "signal-website"]], "FitMS": [[20, "fitms"]], "Causal mediation analysis": [[18, "causal-mediation-analysis"]], "GATK4 mitochondria SNPs and INDELs pipeline": [[14, "gatk4-mitochondria-snps-and-indels-pipeline"]], "Step 0: Download GATK\u2019s public reference databases": [[14, "step-0-download-gatk-s-public-reference-databases"]], "Step 1: Subset a whole genome bam to just Mitochondria reads and remove alignment information": [[14, "step-1-subset-a-whole-genome-bam-to-just-mitochondria-reads-and-remove-alignment-information"]], "Step 2: Align unmapped bam, mark duplicates, collect coverage information, and call SNPs and INDELs from normal bams": [[14, "step-2-align-unmapped-bam-mark-duplicates-collect-coverage-information-and-call-snps-and-indels-from-normal-bams"]], "Step 3: Call SNPs and INDELs from tumor bams": [[14, "step-3-call-snps-and-indels-from-tumor-bams"]], "Step 4: Estimate levels of contamination in mitochondria": [[14, "step-4-estimate-levels-of-contamination-in-mitochondria"]], "GATK4 germline short variant discovery (SNPs + Indels)": [[13, "gatk4-germline-short-variant-discovery-snps-indels"]], "Epigenome-wide association study": [[11, "epigenome-wide-association-study"]]}, "indexentries": {}})