Search.setIndex({"docnames": ["cnv/README", "data_visualization/chord_diagram/README", "data_visualization/heatmap/README", "data_visualization/meta_analysis_forest_plot/README", "data_visualization/meta_analysis_funnel_plot/README", "data_visualization/regression_coefficient_ci/README", "eigenetic_clock/README", "ewas/README", "gatk_data_preprocessing/README", "gatk_germline_snp_indel/README", "gatk_mitochondria_snp_indel/README", "gatk_somatic_snv_indel/README", "index", "mediation_analysis/README", "mqtl/README", "mutational_signatures/README", "sv/README", "tableone/README", "telomere_length/README", "year_one_screening_exam/README"], "filenames": ["cnv/README.md", "data_visualization/chord_diagram/README.md", "data_visualization/heatmap/README.md", "data_visualization/meta_analysis_forest_plot/README.md", "data_visualization/meta_analysis_funnel_plot/README.md", "data_visualization/regression_coefficient_ci/README.md", "eigenetic_clock/README.md", "ewas/README.md", "gatk_data_preprocessing/README.md", "gatk_germline_snp_indel/README.md", "gatk_mitochondria_snp_indel/README.md", "gatk_somatic_snv_indel/README.md", "index.rst", "mediation_analysis/README.md", "mqtl/README.md", "mutational_signatures/README.md", "sv/README.md", "tableone/README.md", "telomere_length/README.md", "year_one_screening_exam/README.md"], "titles": ["CNV calling pipeline", "Chord diagram", "Correlation heatmap", "Meta analysis forest plot", "Generate funnel plot for meta-analysis", "Regression coefficients and CIs", "Eigenetic clock", "Epigenome-wide association study", "Data pre-processing for variant discovery GATK4", "GATK4 germline short variant discovery (SNPs + Indels)", "GATK4 mitochondria SNPs and INDELs pipeline", "GATK4 call somatic snvs indels", "phd-pipeline", "Causal mediation analysis", "Methylation quantitative trait loci analysis", "Mutational Signatures Pipelines", "Calling Structural Variations", "Create summary stats table using R tableone package", "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)", "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam &#64;usc"], "terms": {"thi": [0, 1, 2, 3, 4, 5, 6, 8, 10, 11, 13, 14, 15, 16], "i": [0, 1, 2, 3, 5, 6, 10, 11, 12, 13, 14, 15, 16], "copi": 0, "number": [0, 1, 14, 18], "variat": [0, 11], "from": [0, 5, 6, 7, 9, 15, 16], "76": [0, 16], "match": [0, 8, 10, 11, 16], "tumor": [0, 1, 8, 11, 16], "normal": [0, 8, 11, 13, 16], "whole": [0, 8, 11, 16], "genom": [0, 8, 11, 12, 16], "sequenc": [0, 8, 10, 11, 16], "wg": [0, 8, 9, 10, 11, 16, 18], "bam": [0, 11, 16], "38": [0, 16], "childhood": [0, 8, 10, 11, 14, 16, 18], "acut": [0, 8, 10, 11, 16, 18], "lymphoblast": [0, 8, 10, 11, 16, 18], "leukemia": [0, 8, 10, 11, 16, 18], "case": [0, 7, 8, 10, 11, 16, 18], "file": [0, 8, 15, 18], "have": [0, 11, 15], "been": [0, 15], "data": [0, 6, 10, 11], "check": [0, 8, 9, 10, 11], "out": [0, 8, 9, 10, 11], "best": [0, 9, 10], "practic": [0, 8, 9, 10], "document": [0, 8, 9, 11, 18], "cnv_somatic_pair_workflow": 0, "wdl": [0, 8, 9, 10, 11], "how": [0, 11], "part": 0, "sensit": 0, "detect": [0, 11, 15], "ratio": 0, "alter": [0, 11], "allel": [0, 11], "segment": 0, "ii": [0, 13], "legaci": 0, "tutori": [0, 8], "0": [0, 11, 14], "extractintervalfrompon": 0, "sh": [0, 8, 10, 11, 16], "1": [0, 9, 14, 16, 18], "9": [0, 10], "funcotateseg": 0, "workflow": [0, 6, 7, 9, 10, 11, 12, 13, 14], "A": [0, 6, 7], "figur": 0, "gitc": 0, "2": [0, 14, 18], "cnvkitaccess": 0, "3": [0, 16, 18], "iamp21cnvkitplot": 0, "cnvkittheta": 0, "cnvkitimporttheta": 0, "4": [0, 16, 18], "gisticcnvkitexportseg": 0, "gisticmergeseg": 0, "r": [0, 7, 10, 13, 14, 15], "script": [1, 2, 3, 5, 6, 7, 11, 13, 14, 17], "gener": [1, 7, 17, 18], "show": [1, 2, 5, 13], "gene": 1, "delet": [1, 11], "frequenc": [1, 11], "eight": 1, "commonli": 1, "region": 1, "cdkn2a": 1, "etv6": 1, "ikzf1": 1, "pax5": 1, "btg1": 1, "ebf1": 1, "rb1": 1, "within": [1, 18], "pseudoautosom": 1, "par1": 1, "sex": [1, 7, 13], "chromosom": [1, 7], "crlf2": 1, "csf2ra": 1, "il3ra": 1, "thei": [1, 11], "were": [1, 6, 7, 8, 11, 13, 14, 18], "assai": 1, "dna": [1, 6, 7, 14], "us": [1, 6, 7, 11, 13, 14, 15, 16, 18], "multiplex": 1, "ligat": 1, "depend": [1, 14], "probe": 1, "amplif": 1, "mlpa": 1, "studi": [1, 2, 3, 4, 5, 6, 13, 14, 17], "public": [1, 2, 3, 5, 6, 7, 13, 14], "link": [1, 2, 3, 5, 6, 7, 13, 14, 17], "output": [1, 2, 3, 5, 18], "creat": [2, 3, 5, 11, 12], "pearson": 2, "coeffici": [2, 12], "self": 2, "report": 2, "smoke": 2, "variabl": 2, "fig2": 4, "fig3": 4, "http": [4, 8, 9, 11, 18], "www": 4, "liebertpub": 4, "com": [4, 8, 9, 11, 18], "doi": 4, "ab": 4, "10": [4, 7], "1089": 4, "jicm": 4, "2022": 4, "0586": 4, "plot": [5, 7, 12], "multipl": [5, 11, 14], "calcul": [6, 18], "five": 6, "epigenet": 6, "newborn": 6, "down": 6, "syndrom": 6, "illumina": [6, 8, 18], "methylationep": 6, "methyl": [6, 7, 12, 13], "arrai": 6, "These": 6, "includ": [6, 11, 13, 18], "two": [6, 11, 13, 18], "ag": 6, "dnamskinbloodclock": 6, "pan": 6, "tissu": 6, "dnamag": 6, "three": [6, 13, 14, 16], "gestat": 6, "haftorn": 6, "knight": 6, "bohlin": 6, "base": [6, 10, 13], "estim": [6, 7, 11, 12, 13], "accord": [6, 8, 9, 11], "method": [6, 8], "horvath": 6, "2013": 6, "et": 6, "al": 6, "2018": 6, "skin": 6, "blood": 6, "briefli": 6, "calibr": 6, "version": [6, 8], "wa": [6, 8, 11, 13, 14, 18], "weight": [6, 18], "combin": 6, "level": [6, 11, 13], "353": 6, "cpg": [6, 7, 14], "subsequ": [6, 14], "convert": 6, "back": 6, "similar": 6, "wai": 6, "391": 6, "2016": 6, "2021": 6, "The": [6, 11, 13, 14], "compris": 6, "176": 6, "epic": 6, "ar": [6, 11], "compos": 6, "148": 6, "96": 6, "450k": 6, "respect": 6, "six": 6, "8": [6, 10], "absent": 6, "thu": 6, "replac": 6, "averag": [6, 18], "beta": 6, "valu": 6, "142": 6, "88": 6, "avail": [6, 11], "imput": 6, "knn": 6, "function": [6, 7, 13, 14], "full": 6, "list": 6, "can": [6, 11], "found": 6, "here": 6, "run": [7, 8, 9, 18], "ewa": 7, "analys": [7, 13], "identifi": [7, 14], "all": [7, 13, 14, 18], "differenti": 7, "posit": 7, "autosom": 7, "we": [7, 11, 13, 14, 18], "fit": [7, 13, 14], "logist": 7, "regress": [7, 12, 14], "model": [7, 11, 13, 14], "predict": [7, 14], "control": 7, "statu": [7, 13], "each": [7, 11, 14, 18], "remain": 7, "adjust": [7, 14], "batch": [7, 13], "effect": [7, 11, 13, 14], "cell": [7, 13], "type": [7, 13], "heterogen": [7, 13], "first": [7, 13, 14], "princip": 7, "compon": 7, "deriv": 7, "refactor": 7, "58": 7, "genet": 7, "ancestri": [7, 13], "epistructur": 7, "fix": [7, 13, 14], "meta": [7, 12, 13, 14], "analysi": [7, 12], "summari": [7, 12, 16], "result": [7, 13, 14, 18], "overlap": 7, "hree": 7, "differ": [7, 11, 15], "set": [7, 11], "packag": [7, 13, 14], "metafor": [7, 14], "bidirect": 7, "manhattan": 7, "qq": 7, "sampl": [7, 11, 17, 18], "cmplot": 7, "rmd": [7, 8, 15], "an": [8, 9, 10, 11, 13, 14, 16, 18], "hpc": [8, 9, 10, 11, 16], "50": [8, 11], "25": [8, 10, 11, 18], "novaseq": [8, 18], "6000": [8, 18], "gatk": [8, 9, 11, 15, 18], "doc": [8, 10, 11], "more": [8, 9, 10, 11], "detail": [8, 9, 10, 11], "written": [8, 11], "follow": [8, 11], "offici": [8, 11], "github": [8, 9, 11, 18], "pipelin": [8, 9, 11, 16, 18], "blob": [8, 9], "master": [8, 9, 11], "task": [8, 9], "unmappedbamtoalignedbam": 8, "wholegenomegermlinesinglesampl": 8, "simpl": 8, "broadgsa": 8, "doc_arch": 8, "reference_implement": 8, "_pairedendsinglesamplewf_pipelin": 8, "md": 8, "sortandfixtag": 8, "other": [8, 11], "introduct": 8, "biowulf": 8, "nih": 8, "howto": 8, "bqsr": 8, "fastqc": 8, "pairedfastqtounmappedbam": 8, "validatesamfil": 8, "sortsam": 8, "bwa": 8, "sortalignedbam": 8, "validatealignedbam": 8, "checkifaltawarealign": 8, "mergebamalign": 8, "validatemergedbam": 8, "countread": 8, "validatesortedmarkedbam": 8, "baserecalibr": 8, "bqsrplot": 8, "haplotypecal": 9, "gvcf": 9, "mode": [9, 11], "broad": 9, "prod": 9, "jointgenotypingwf": 9, "broadinstitut": [9, 11], "warp": 9, "jointgenotypingtask": 9, "dna_seq": 9, "joint_genotyp": 9, "jointgenotyp": 9, "mitochondri": [10, 12], "larg": 10, "subsetbamtochrm": 10, "revertsam": 10, "aligntomt": 10, "aligntoshiftedmt": 10, "collectwgsmetr": 10, "runcollectwgsmetricsr": 10, "callmt": 10, "germlin": [10, 11, 12], "5": [10, 15, 16], "callshiftedmt": 10, "6": [10, 11], "liftoverandcombinevcf": 10, "7": [10, 18], "mergestat": [10, 11], "initialfilt": 10, "splitmultiallelicsandremovenonpasssit": 10, "getcontamin": 10, "filtercontamin": 10, "mutat": [11, 12], "singl": 11, "nucleotid": 11, "sna": 11, "insert": 11, "preprocess": 11, "tree": 11, "mutect2_wdl": 11, "explan": 11, "deprec": 11, "new": 11, "pdf": 11, "featur": 11, "improv": 11, "what": 11, "tradit": 11, "entail": 11, "faq": 11, "onli": 11, "versu": 11, "doe": 11, "do": 11, "In": 11, "popul": 11, "resourc": 11, "prior": 11, "read": [11, 18], "fraction": 11, "\u00bd": 11, "suggest": 11, "het": 11, "evid": 11, "miss": 11, "": 11, "same": [11, 13], "addition": 11, "power": 11, "exclud": 11, "rare": 11, "captur": 11, "individu": [11, 18], "specif": 11, "toward": 11, "being": 11, "simplifi": 11, "site": [11, 14], "gnomad": 11, "retain": 11, "ftp": 11, "gsapubftp": 11, "anonym": 11, "org": 11, "bundl": 11, "panel": 11, "pon": 11, "ha": 11, "vital": 11, "role": 11, "fill": 11, "gap": 11, "between": [11, 14], "catch": 11, "addit": [11, 14], "nois": 11, "like": 11, "map": 11, "somewhat": 11, "random": 11, "systemat": 11, "process": [11, 12, 15, 18], "interest": 11, "biostar": 11, "post": 11, "heterozygot": 11, "And": 11, "manifest": 11, "project": [11, 12], "contain": 11, "artifactu": 11, "tool": 11, "take": [11, 18], "callset": 11, "produc": 11, "collat": 11, "present": 11, "default": 11, "min": 11, "count": 11, "argument": 11, "vcf": [11, 15], "common": 11, "max": 11, "probabl": 11, "threshold": [11, 14, 18], "possibl": 11, "By": 11, "so": 11, "event": 11, "usual": 11, "correct": [11, 14], "behavior": 11, "handl": 11, "probabilist": 11, "via": 11, "human": 11, "much": 11, "refin": 11, "than": 11, "ani": 11, "could": 11, "final": [11, 18], "tabl": [11, 12, 18], "tell": 11, "u": 11, "rate": 11, "cross": 11, "If": 11, "15": 11, "error": 11, "19": 11, "mean": 11, "one": [11, 12, 17], "everi": 11, "hundr": 11, "come": 11, "someon": 11, "els": 11, "percentag": 11, "our": 11, "floor": 11, "low": 11, "potenti": 11, "below": 11, "af": 11, "would": 11, "zero": 11, "judg": 11, "whether": 11, "real": 11, "through": 11, "feed": 11, "properli": 11, "account": 11, "scatterinterv": 11, "mutect2scattergather0": 11, "mergevcf": 11, "mergebamout": 11, "orientationbia": 11, "createbwaimag": 11, "filteralignmentartifact": 11, "funcot": 11, "bcftoolfilt": 11, "some": 12, "wrote": 12, "my": 12, "research": 12, "pre": [12, 18], "variant": [12, 18], "discoveri": [12, 18], "gatk4": 12, "short": 12, "snp": [12, 14], "indel": 12, "call": 12, "somat": 12, "snv": 12, "sv": 12, "cnv": 12, "signatur": 12, "telomer": 12, "length": 12, "eigenet": 12, "clock": 12, "epigenom": 12, "wide": 12, "quantit": 12, "trait": 12, "loci": 12, "causal": 12, "mediat": 12, "stat": 12, "chord": 12, "diagram": [12, 13], "correl": 12, "heatmap": 12, "ci": 12, "funnel": 12, "forest": 12, "year": 12, "screen": 12, "exam": 12, "perform": 13, "signific": 13, "mqtl": [13, 14], "dmp": [13, 14], "pair": 13, "specifi": 13, "statist": 13, "distribut": 13, "after": 13, "condit": 13, "genotyp": [13, 14], "covari": 13, "outcom": 13, "given": 13, "separ": [13, 14], "paramet": 13, "main": 13, "input": 13, "which": 13, "comput": 13, "acm": 13, "ad": 13, "total": 13, "varianc": 13, "simul": 13, "quasi": 13, "bayesian": 13, "mont": 13, "carlo": 13, "approxim": 13, "conduct": [13, 14], "1000": 13, "time": 13, "altern": 13, "approach": 13, "non": 13, "parametr": 13, "bootstrap": 13, "also": 13, "appli": [13, 18], "valid": 13, "dataset": [13, 14], "summar": 13, "across": [13, 14], "path": 13, "confound": 13, "associ": 14, "risk": 14, "matrix": 14, "eqtl": 14, "linear": 14, "code": [14, 19], "test": 14, "stringent": 14, "bonferroni": 14, "025": 14, "fdr": 14, "05": 14, "analyz": 14, "mutect2": 15, "algorithm": 15, "step1": 15, "py": 15, "step2": 15, "step3": 15, "step4": 15, "signalsummari": 15, "signalkataegi": 15, "step5": 15, "softwar": [16, 18], "premergesvpersamplesmoov": 16, "smooveunionsit": 16, "smoovegenotyp": 16, "smoovepasteannot": 16, "premergesvpersamplemantaconfigur": 16, "mantaworkflow": 16, "mantaclean": 16, "dellypersamplecal": 16, "dellyprefilt": 16, "dellygeno": 16, "dellypostfilt": 16, "dellybcf2vcf": 16, "survivor": 16, "anno": 16, "exampl": 16, "graph": 16, "inform": 17, "compar": 18, "paper": 18, "content": 18, "quantifi": 18, "ten": 18, "repeat": 18, "tcaggg": 18, "tgaggg": 18, "ttgggg": 18, "ttcggg": 18, "tttggg": 18, "ataggg": 18, "catggg": 18, "ctaggg": 18, "gtaggg": 18, "taaggg": 18, "tl": 18, "wsa": 18, "kb": 18, "ttaggg": 18, "ccctaa": 18, "consid": 18, "step": 18, "aggreg": 18, "clean": 18, "need": 18, "group": 18, "zd1": 18, "issu": 18, "had": 18, "high": 18}, "objects": {}, "objtypes": {}, "objnames": {}, "titleterms": {"cnv": 0, "call": [0, 10, 11, 16], "pipelin": [0, 10, 12, 15], "gatk": [0, 10], "somat": [0, 11], "cnvkit": 0, "preprocess": [0, 8], "creat": [0, 16, 17], "theta": 0, "input": 0, "gistic": 0, "chord": 1, "diagram": [1, 16, 19], "correl": [2, 18], "heatmap": 2, "meta": [3, 4], "analysi": [3, 4, 13, 14], "forest": 3, "plot": [3, 4, 18], "gener": 4, "funnel": 4, "regress": [5, 19], "coeffici": 5, "ci": 5, "eigenet": 6, "clock": 6, "epigenom": 7, "wide": 7, "associ": [7, 12], "studi": [7, 12], "data": [8, 12, 18], "pre": 8, "process": 8, "variant": [8, 9, 11, 16], "discoveri": [8, 9, 11], "gatk4": [8, 9, 10, 11], "The": 8, "main": 8, "step": [8, 10, 11], "workflow": [8, 18], "A": [8, 11, 18], "figur": [8, 11], "from": [8, 10, 11, 18], "gitc": [8, 11], "0": [8, 10], "qc": 8, "1": [8, 10, 11], "fastq": 8, "unmap": [8, 10], "bam": [8, 10, 18], "2": [8, 10, 11, 16], "map": 8, "3": [8, 10, 11], "merg": 8, "4": [8, 10, 11], "markdupl": 8, "5": [8, 11], "base": 8, "qualiti": 8, "score": 8, "recalibr": 8, "germlin": 9, "short": [9, 11], "snp": [9, 10], "indel": [9, 10, 11], "mitochondria": 10, "download": 10, "": 10, "public": 10, "refer": 10, "databas": 10, "subset": 10, "whole": [10, 18], "genom": [10, 18], "just": 10, "read": 10, "remov": 10, "align": [10, 11], "inform": 10, "mark": 10, "duplic": 10, "collect": 10, "coverag": 10, "normal": [10, 18], "tumor": [10, 18], "estim": [10, 18, 19], "level": 10, "contamin": [10, 11], "snv": 11, "best": 11, "practic": 11, "scatter": 11, "interv": 11, "mutect2": 11, "parallel": 11, "gather": 11, "result": 11, "calcul": 11, "learn": 11, "orient": 11, "bia": 11, "filter": 11, "artifact": 11, "phd": [12, 19], "bioinformat": 12, "visual": 12, "other": 12, "causal": 13, "mediat": 13, "methyl": 14, "quantit": 14, "trait": 14, "loci": 14, "mutat": 15, "signatur": 15, "sigprofil": 15, "pmsignatur": 15, "deconstructsig": 15, "signal": 15, "websit": 15, "fitm": 15, "structur": 16, "variat": 16, "smoov": 16, "lumpi": 16, "manta": 16, "delli": 16, "find": 16, "were": 16, "least": 16, "caller": 16, "annot": 16, "venn": 16, "histogram": 16, "html": 16, "file": 16, "summari": 17, "stat": 17, "tabl": 17, "us": 17, "r": 17, "tableon": 17, "packag": 17, "telomer": 18, "length": 18, "match": 18, "sequenc": 18, "background": 18, "telomerehunt": 18, "telseq": 18, "exampl": 18, "sa": 19, "script": 19, "spider": 19, "i": 19, "prepar": 19, "epidemiologi": 19, "screen": 19, "exam": 19, "usc": 19, "appli": 19, "predict": 19, "theori": 19}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinx.ext.viewcode": 1, "sphinx": 57}, "alltitles": {"CNV calling pipeline": [[0, "cnv-calling-pipeline"]], "GATK somatic CNV calling pipeline:": [[0, "gatk-somatic-cnv-calling-pipeline"]], "CNVkit pipeline": [[0, "cnvkit-pipeline"]], "Preprocessing to create Theta inputs": [[0, "preprocessing-to-create-theta-inputs"]], "Gistic pipeline": [[0, "gistic-pipeline"]], "Data pre-processing for variant discovery GATK4": [[8, "data-pre-processing-for-variant-discovery-gatk4"]], "The main steps in the preprocessing workflow. A figure from GITC.": [[8, "the-main-steps-in-the-preprocessing-workflow-a-figure-from-gitc"]], "Step 0: QC": [[8, "step-0-qc"]], "Step 1: Fastqs to Unmapped BAMs": [[8, "step-1-fastqs-to-unmapped-bams"]], "Step 2: Unmapped BAMs to Mapped BAMs": [[8, "step-2-unmapped-bams-to-mapped-bams"]], "Step 3: Merge Unmapped BAMs and Mapped BAMs": [[8, "step-3-merge-unmapped-bams-and-mapped-bams"]], "Step 4: MarkDuplicates": [[8, "step-4-markduplicates"]], "Step 5: Base Quality Score Recalibration": [[8, "step-5-base-quality-score-recalibration"]], "Chord diagram": [[1, "chord-diagram"]], "Generate funnel plot for meta-analysis": [[4, "generate-funnel-plot-for-meta-analysis"]], "Regression coefficients and CIs": [[5, "regression-coefficients-and-cis"]], "Eigenetic clock": [[6, "eigenetic-clock"]], "Epigenome-wide association study": [[7, "epigenome-wide-association-study"]], "Correlation heatmap": [[2, "correlation-heatmap"]], "Meta analysis forest plot": [[3, "meta-analysis-forest-plot"]], "GATK4 germline short variant discovery (SNPs + Indels)": [[9, "gatk4-germline-short-variant-discovery-snps-indels"]], "Create summary stats table using R tableone package": [[17, "create-summary-stats-table-using-r-tableone-package"]], "GATK4 mitochondria SNPs and INDELs pipeline": [[10, "gatk4-mitochondria-snps-and-indels-pipeline"]], "Step 0: Download GATK\u2019s public reference databases": [[10, "step-0-download-gatk-s-public-reference-databases"]], "Step 1: Subset a whole genome bam to just Mitochondria reads and remove alignment information": [[10, "step-1-subset-a-whole-genome-bam-to-just-mitochondria-reads-and-remove-alignment-information"]], "Step 2: Align unmapped bam, mark duplicates, collect coverage information, and call SNPs and INDELs from normal bams": [[10, "step-2-align-unmapped-bam-mark-duplicates-collect-coverage-information-and-call-snps-and-indels-from-normal-bams"]], "Step 3: Call SNPs and INDELs from tumor bams": [[10, "step-3-call-snps-and-indels-from-tumor-bams"]], "Step 4: Estimate levels of contamination in mitochondria": [[10, "step-4-estimate-levels-of-contamination-in-mitochondria"]], "Calling Structural Variations": [[16, "calling-structural-variations"]], "Smoove (lumpy)": [[16, "smoove-lumpy"]], "Manta": [[16, "manta"]], "Delly": [[16, "delly"]], "Finding variants that were called by at least 2 callers": [[16, "finding-variants-that-were-called-by-at-least-2-callers"]], "Annotate variants": [[16, "annotate-variants"]], "Create Venn diagram, histograms and html files": [[16, "create-venn-diagram-histograms-and-html-files"]], "Causal mediation analysis": [[13, "causal-mediation-analysis"]], "Methylation quantitative trait loci analysis": [[14, "methylation-quantitative-trait-loci-analysis"]], "Mutational Signatures Pipelines": [[15, "mutational-signatures-pipelines"]], "Sigprofiler": [[15, "sigprofiler"]], "pmsignature": [[15, "pmsignature"]], "deconstructSigs": [[15, "deconstructsigs"]], "Signal website": [[15, "signal-website"]], "FitMS": [[15, "fitms"]], "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam @usc": [[19, "sas-scripts-and-spider-diagram-that-i-prepared-for-the-epidemiology-phd-screening-exam-usc"]], "apply exam - regression estimation and prediction": [[19, "apply-exam-regression-estimation-and-prediction"]], "theory - epidemiology": [[19, "theory-epidemiology"]], "phd-pipeline": [[12, "phd-pipeline"]], "Bioinformatics pipelines": [[12, "bioinformatics-pipelines"]], "Association studies": [[12, "association-studies"]], "Data visualization": [[12, "data-visualization"]], "Other": [[12, "other"]], "GATK4 call somatic snvs indels": [[11, "gatk4-call-somatic-snvs-indels"]], "Best practices for somatic short variant discovery. A figure from GITC.": [[11, "best-practices-for-somatic-short-variant-discovery-a-figure-from-gitc"]], "Step 1: Scatter interval for Mutect2 parallel calling": [[11, "step-1-scatter-interval-for-mutect2-parallel-calling"]], "Step 2: Gather results from scattered intervals": [[11, "step-2-gather-results-from-scattered-intervals"]], "Step 3: Calculate contamination and learn orientation bias": [[11, "step-3-calculate-contamination-and-learn-orientation-bias"]], "Step 4: Filter alignment artifacts": [[11, "step-4-filter-alignment-artifacts"]], "Step 5: Filter variants": [[11, "step-5-filter-variants"]], "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)": [[18, "a-workflow-to-estimate-telomere-length-from-matched-tumor-normal-whole-genome-sequencing-data-bams"]], "Background": [[18, "background"]], "TelomereHunter": [[18, "telomerehunter"]], "TelSeq": [[18, "telseq"]], "Workflow": [[18, "workflow"]], "Example correlation plot": [[18, "example-correlation-plot"]]}, "indexentries": {}})