Search.setIndex({"docnames": ["cnv/README", "data_visualization/chord_diagram/README", "data_visualization/heatmap/README", "data_visualization/meta_analysis_forest_plot/README", "data_visualization/meta_analysis_funnel_plot/README", "data_visualization/regression_coefficient_ci/README", "ewas/README", "gatk_data_preprocessing/README", "gatk_germline_snp_indel/README", "gatk_mitochondria_snp_indel/README", "gatk_somatic_snv_indel/README", "index", "mediation_analysis/README", "mqtl/README", "mutational_signatures/README", "sv/README", "tableone/README", "telomere_length/README", "year_one_screening_exam/README"], "filenames": ["cnv/README.md", "data_visualization/chord_diagram/README.md", "data_visualization/heatmap/README.md", "data_visualization/meta_analysis_forest_plot/README.md", "data_visualization/meta_analysis_funnel_plot/README.md", "data_visualization/regression_coefficient_ci/README.md", "ewas/README.md", "gatk_data_preprocessing/README.md", "gatk_germline_snp_indel/README.md", "gatk_mitochondria_snp_indel/README.md", "gatk_somatic_snv_indel/README.md", "index.rst", "mediation_analysis/README.md", "mqtl/README.md", "mutational_signatures/README.md", "sv/README.md", "tableone/README.md", "telomere_length/README.md", "year_one_screening_exam/README.md"], "titles": ["CNV calling pipeline", "Chord diagram", "Correlation heatmap", "Meta analysis forest plot", "Generate funnel plot for meta-analysis", "Regression coefficients and CIs", "Epigenome-wide association study", "Data pre-processing for variant discovery GATK4", "GATK4 germline short variant discovery (SNPs + Indels)", "GATK4 mitochondria SNPs and INDELs pipeline", "GATK4 call somatic snvs indels", "phd-pipeline", "Causal mediation analysis", "Methylation quantitative trait loci analysis", "Mutational Signatures Pipelines", "Calling Structural Variations", "Create summary stats table using R tableone package", "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)", "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam &#64;usc"], "terms": {"thi": [0, 1, 2, 3, 4, 5, 7, 9, 10, 12, 13, 14, 15], "i": [0, 1, 2, 3, 5, 9, 10, 11, 12, 13, 14, 15], "copi": 0, "number": [0, 1, 13, 17], "variat": [0, 10], "from": [0, 5, 6, 8, 14, 15], "76": [0, 15], "match": [0, 7, 9, 10, 15], "tumor": [0, 1, 7, 10, 15], "normal": [0, 7, 10, 12, 15], "whole": [0, 7, 10, 15], "genom": [0, 7, 10, 11, 15], "sequenc": [0, 7, 9, 10, 15], "wg": [0, 7, 8, 9, 10, 15, 17], "bam": [0, 10, 15], "38": [0, 15], "childhood": [0, 7, 9, 10, 13, 15, 17], "acut": [0, 7, 9, 10, 15, 17], "lymphoblast": [0, 7, 9, 10, 15, 17], "leukemia": [0, 7, 9, 10, 15, 17], "case": [0, 6, 7, 9, 10, 15, 17], "file": [0, 7, 14, 17], "have": [0, 10, 14], "been": [0, 14], "data": [0, 9, 10], "check": [0, 7, 8, 9, 10], "out": [0, 7, 8, 9, 10], "best": [0, 8, 9], "practic": [0, 7, 8, 9], "document": [0, 7, 8, 10, 17], "cnv_somatic_pair_workflow": 0, "wdl": [0, 7, 8, 9, 10], "how": [0, 10], "part": 0, "sensit": 0, "detect": [0, 10, 14], "ratio": 0, "alter": [0, 10], "allel": [0, 10], "segment": 0, "ii": [0, 12], "legaci": 0, "tutori": [0, 7], "0": [0, 10, 13], "extractintervalfrompon": 0, "sh": [0, 7, 9, 10, 15], "1": [0, 8, 13, 15, 17], "9": [0, 9], "funcotateseg": 0, "workflow": [0, 6, 8, 9, 10, 11, 12, 13], "A": [0, 6], "figur": 0, "gitc": 0, "2": [0, 13, 17], "cnvkitaccess": 0, "3": [0, 15, 17], "iamp21cnvkitplot": 0, "cnvkittheta": 0, "cnvkitimporttheta": 0, "4": [0, 15, 17], "gisticcnvkitexportseg": 0, "gisticmergeseg": 0, "r": [0, 6, 9, 12, 13, 14], "script": [1, 2, 3, 5, 6, 10, 12, 13, 16], "gener": [1, 6, 16, 17], "show": [1, 2, 5, 12], "gene": 1, "delet": [1, 10], "frequenc": [1, 10], "eight": 1, "commonli": 1, "region": 1, "cdkn2a": 1, "etv6": 1, "ikzf1": 1, "pax5": 1, "btg1": 1, "ebf1": 1, "rb1": 1, "within": [1, 17], "pseudoautosom": 1, "par1": 1, "sex": [1, 6, 12], "chromosom": [1, 6], "crlf2": 1, "csf2ra": 1, "il3ra": 1, "thei": [1, 10], "were": [1, 6, 7, 10, 12, 13, 17], "assai": 1, "dna": [1, 6, 13], "us": [1, 6, 10, 12, 13, 14, 15, 17], "multiplex": 1, "ligat": 1, "depend": [1, 13], "probe": 1, "amplif": 1, "mlpa": 1, "studi": [1, 2, 3, 4, 5, 12, 13, 16], "public": [1, 2, 3, 5, 6, 12, 13], "link": [1, 2, 3, 5, 6, 12, 13, 16], "output": [1, 2, 3, 5, 17], "creat": [2, 3, 5, 10, 11], "pearson": 2, "coeffici": [2, 11], "self": 2, "report": 2, "smoke": 2, "variabl": 2, "fig2": 4, "fig3": 4, "http": [4, 7, 8, 10, 17], "www": 4, "liebertpub": 4, "com": [4, 7, 8, 10, 17], "doi": 4, "ab": 4, "10": [4, 6], "1089": 4, "jicm": 4, "2022": 4, "0586": 4, "plot": [5, 6, 11], "multipl": [5, 10, 13], "run": [6, 7, 8, 17], "ewa": 6, "analys": [6, 12], "identifi": [6, 13], "all": [6, 12, 13, 17], "differenti": 6, "methyl": [6, 11, 12], "cpg": [6, 13], "posit": 6, "autosom": 6, "we": [6, 10, 12, 13, 17], "fit": [6, 12, 13], "logist": 6, "regress": [6, 11, 13], "model": [6, 10, 12, 13], "predict": [6, 13], "control": 6, "statu": [6, 12], "function": [6, 12, 13], "each": [6, 10, 13, 17], "remain": 6, "adjust": [6, 13], "batch": [6, 12], "effect": [6, 10, 12, 13], "cell": [6, 12], "type": [6, 12], "heterogen": [6, 12], "first": [6, 12, 13], "princip": 6, "compon": 6, "deriv": 6, "refactor": 6, "58": 6, "genet": 6, "ancestri": [6, 12], "epistructur": 6, "fix": [6, 12, 13], "meta": [6, 11, 12, 13], "analysi": [6, 11], "summari": [6, 11, 15], "estim": [6, 10, 11, 12], "result": [6, 12, 13, 17], "overlap": 6, "hree": 6, "differ": [6, 10, 14], "set": [6, 10], "packag": [6, 12, 13], "metafor": [6, 13], "bidirect": 6, "manhattan": 6, "qq": 6, "sampl": [6, 10, 16, 17], "cmplot": 6, "rmd": [6, 7, 14], "an": [7, 8, 9, 10, 12, 13, 15, 17], "hpc": [7, 8, 9, 10, 15], "50": [7, 10], "25": [7, 9, 10, 17], "illumina": [7, 17], "novaseq": [7, 17], "6000": [7, 17], "gatk": [7, 8, 10, 14, 17], "doc": [7, 9, 10], "more": [7, 8, 9, 10], "detail": [7, 8, 9, 10], "wa": [7, 10, 12, 13, 17], "written": [7, 10], "accord": [7, 8, 10], "follow": [7, 10], "offici": [7, 10], "version": 7, "github": [7, 8, 10, 17], "pipelin": [7, 8, 10, 15, 17], "blob": [7, 8], "master": [7, 8, 10], "task": [7, 8], "unmappedbamtoalignedbam": 7, "wholegenomegermlinesinglesampl": 7, "simpl": 7, "broadgsa": 7, "doc_arch": 7, "method": 7, "reference_implement": 7, "_pairedendsinglesamplewf_pipelin": 7, "md": 7, "sortandfixtag": 7, "other": [7, 10], "introduct": 7, "biowulf": 7, "nih": 7, "howto": 7, "bqsr": 7, "fastqc": 7, "pairedfastqtounmappedbam": 7, "validatesamfil": 7, "sortsam": 7, "bwa": 7, "sortalignedbam": 7, "validatealignedbam": 7, "checkifaltawarealign": 7, "mergebamalign": 7, "validatemergedbam": 7, "countread": 7, "validatesortedmarkedbam": 7, "baserecalibr": 7, "bqsrplot": 7, "haplotypecal": 8, "gvcf": 8, "mode": [8, 10], "broad": 8, "prod": 8, "jointgenotypingwf": 8, "broadinstitut": [8, 10], "warp": 8, "jointgenotypingtask": 8, "dna_seq": 8, "joint_genotyp": 8, "jointgenotyp": 8, "mitochondri": [9, 11], "larg": 9, "base": [9, 12], "subsetbamtochrm": 9, "revertsam": 9, "aligntomt": 9, "aligntoshiftedmt": 9, "collectwgsmetr": 9, "runcollectwgsmetricsr": 9, "callmt": 9, "germlin": [9, 10, 11], "5": [9, 14, 15], "callshiftedmt": 9, "6": [9, 10], "liftoverandcombinevcf": 9, "7": [9, 17], "mergestat": [9, 10], "8": 9, "initialfilt": 9, "splitmultiallelicsandremovenonpasssit": 9, "getcontamin": 9, "filtercontamin": 9, "mutat": [10, 11], "includ": [10, 12, 17], "singl": 10, "nucleotid": 10, "sna": 10, "insert": 10, "preprocess": 10, "tree": 10, "mutect2_wdl": 10, "explan": 10, "deprec": 10, "new": 10, "pdf": 10, "featur": 10, "improv": 10, "what": 10, "tradit": 10, "entail": 10, "faq": 10, "onli": 10, "versu": 10, "doe": 10, "do": 10, "In": 10, "popul": 10, "resourc": 10, "prior": 10, "read": [10, 17], "fraction": 10, "\u00bd": 10, "ar": 10, "suggest": 10, "het": 10, "evid": 10, "miss": 10, "": 10, "same": [10, 12], "addition": 10, "power": 10, "avail": 10, "exclud": 10, "rare": 10, "captur": 10, "individu": [10, 17], "specif": 10, "toward": 10, "being": 10, "The": [10, 12, 13], "simplifi": 10, "site": [10, 13], "gnomad": 10, "retain": 10, "ftp": 10, "gsapubftp": 10, "anonym": 10, "org": 10, "bundl": 10, "panel": 10, "pon": 10, "ha": 10, "vital": 10, "role": 10, "fill": 10, "gap": 10, "between": [10, 13], "catch": 10, "addit": [10, 13], "nois": 10, "like": 10, "map": 10, "somewhat": 10, "random": 10, "systemat": 10, "process": [10, 11, 14, 17], "interest": 10, "biostar": 10, "post": 10, "heterozygot": 10, "And": 10, "manifest": 10, "project": [10, 11], "contain": 10, "artifactu": 10, "tool": 10, "take": [10, 17], "callset": 10, "produc": 10, "collat": 10, "present": 10, "two": [10, 12, 17], "default": 10, "min": 10, "count": 10, "argument": 10, "vcf": [10, 14], "common": 10, "level": [10, 12], "max": 10, "probabl": 10, "threshold": [10, 13, 17], "possibl": 10, "By": 10, "so": 10, "event": 10, "usual": 10, "correct": [10, 13], "behavior": 10, "handl": 10, "probabilist": 10, "via": 10, "human": 10, "much": 10, "refin": 10, "than": 10, "ani": 10, "could": 10, "final": [10, 17], "tabl": [10, 11, 17], "tell": 10, "u": 10, "rate": 10, "cross": 10, "If": 10, "15": 10, "error": 10, "19": 10, "mean": 10, "one": [10, 11, 16], "everi": 10, "hundr": 10, "come": 10, "someon": 10, "els": 10, "percentag": 10, "our": 10, "floor": 10, "low": 10, "potenti": 10, "below": 10, "af": 10, "would": 10, "zero": 10, "judg": 10, "whether": 10, "real": 10, "through": 10, "feed": 10, "can": 10, "properli": 10, "account": 10, "scatterinterv": 10, "mutect2scattergather0": 10, "mergevcf": 10, "mergebamout": 10, "orientationbia": 10, "createbwaimag": 10, "filteralignmentartifact": 10, "funcot": 10, "bcftoolfilt": 10, "some": 11, "wrote": 11, "my": 11, "research": 11, "pre": [11, 17], "variant": [11, 17], "discoveri": [11, 17], "gatk4": 11, "short": 11, "snp": [11, 13], "indel": 11, "call": 11, "somat": 11, "snv": 11, "sv": 11, "cnv": 11, "signatur": 11, "telomer": 11, "length": 11, "eigenet": 11, "clock": 11, "epigenom": 11, "wide": 11, "quantit": 11, "trait": 11, "loci": 11, "causal": 11, "mediat": 11, "stat": 11, "chord": 11, "diagram": [11, 12], "correl": 11, "heatmap": 11, "ci": 11, "funnel": 11, "forest": 11, "year": 11, "screen": 11, "exam": 11, "perform": 12, "signific": 12, "mqtl": [12, 13], "dmp": [12, 13], "pair": 12, "specifi": 12, "statist": 12, "distribut": 12, "after": 12, "condit": 12, "genotyp": [12, 13], "covari": 12, "outcom": 12, "given": 12, "separ": [12, 13], "paramet": 12, "main": 12, "input": 12, "which": 12, "comput": 12, "acm": 12, "ad": 12, "total": 12, "varianc": 12, "simul": 12, "quasi": 12, "bayesian": 12, "mont": 12, "carlo": 12, "approxim": 12, "conduct": [12, 13], "1000": 12, "time": 12, "altern": 12, "approach": 12, "non": 12, "parametr": 12, "bootstrap": 12, "also": 12, "appli": [12, 17], "valid": 12, "three": [12, 13, 15], "dataset": [12, 13], "summar": 12, "across": [12, 13], "path": 12, "confound": 12, "associ": 13, "risk": 13, "matrix": 13, "eqtl": 13, "linear": 13, "code": [13, 18], "test": 13, "stringent": 13, "bonferroni": 13, "025": 13, "fdr": 13, "05": 13, "subsequ": 13, "analyz": 13, "mutect2": 14, "algorithm": 14, "step1": 14, "py": 14, "step2": 14, "step3": 14, "step4": 14, "signalsummari": 14, "signalkataegi": 14, "step5": 14, "softwar": [15, 17], "premergesvpersamplesmoov": 15, "smooveunionsit": 15, "smoovegenotyp": 15, "smoovepasteannot": 15, "premergesvpersamplemantaconfigur": 15, "mantaworkflow": 15, "mantaclean": 15, "dellypersamplecal": 15, "dellyprefilt": 15, "dellygeno": 15, "dellypostfilt": 15, "dellybcf2vcf": 15, "survivor": 15, "anno": 15, "exampl": 15, "graph": 15, "inform": 16, "compar": 17, "paper": 17, "content": 17, "quantifi": 17, "ten": 17, "repeat": 17, "tcaggg": 17, "tgaggg": 17, "ttgggg": 17, "ttcggg": 17, "tttggg": 17, "ataggg": 17, "catggg": 17, "ctaggg": 17, "gtaggg": 17, "taaggg": 17, "tl": 17, "wsa": 17, "kb": 17, "ttaggg": 17, "ccctaa": 17, "consid": 17, "step": 17, "aggreg": 17, "clean": 17, "calcul": 17, "need": 17, "weight": 17, "averag": 17, "group": 17, "zd1": 17, "issu": 17, "had": 17, "high": 17}, "objects": {}, "objtypes": {}, "objnames": {}, "titleterms": {"cnv": 0, "call": [0, 9, 10, 15], "pipelin": [0, 9, 11, 14], "gatk": [0, 9], "somat": [0, 10], "cnvkit": 0, "preprocess": [0, 7], "creat": [0, 15, 16], "theta": 0, "input": 0, "gistic": 0, "chord": 1, "diagram": [1, 15, 18], "correl": [2, 17], "heatmap": 2, "meta": [3, 4], "analysi": [3, 4, 12, 13], "forest": 3, "plot": [3, 4, 17], "gener": 4, "funnel": 4, "regress": [5, 18], "coeffici": 5, "ci": 5, "epigenom": 6, "wide": 6, "associ": [6, 11], "studi": [6, 11], "data": [7, 11, 17], "pre": 7, "process": 7, "variant": [7, 8, 10, 15], "discoveri": [7, 8, 10], "gatk4": [7, 8, 9, 10], "The": 7, "main": 7, "step": [7, 9, 10], "workflow": [7, 17], "A": [7, 10, 17], "figur": [7, 10], "from": [7, 9, 10, 17], "gitc": [7, 10], "0": [7, 9], "qc": 7, "1": [7, 9, 10], "fastq": 7, "unmap": [7, 9], "bam": [7, 9, 17], "2": [7, 9, 10, 15], "map": 7, "3": [7, 9, 10], "merg": 7, "4": [7, 9, 10], "markdupl": 7, "5": [7, 10], "base": 7, "qualiti": 7, "score": 7, "recalibr": 7, "germlin": 8, "short": [8, 10], "snp": [8, 9], "indel": [8, 9, 10], "mitochondria": 9, "download": 9, "": 9, "public": 9, "refer": 9, "databas": 9, "subset": 9, "whole": [9, 17], "genom": [9, 17], "just": 9, "read": 9, "remov": 9, "align": [9, 10], "inform": 9, "mark": 9, "duplic": 9, "collect": 9, "coverag": 9, "normal": [9, 17], "tumor": [9, 17], "estim": [9, 17, 18], "level": 9, "contamin": [9, 10], "snv": 10, "best": 10, "practic": 10, "scatter": 10, "interv": 10, "mutect2": 10, "parallel": 10, "gather": 10, "result": 10, "calcul": 10, "learn": 10, "orient": 10, "bia": 10, "filter": 10, "artifact": 10, "phd": [11, 18], "bioinformat": 11, "visual": 11, "other": 11, "causal": 12, "mediat": 12, "methyl": 13, "quantit": 13, "trait": 13, "loci": 13, "mutat": 14, "signatur": 14, "sigprofil": 14, "pmsignatur": 14, "deconstructsig": 14, "signal": 14, "websit": 14, "fitm": 14, "structur": 15, "variat": 15, "smoov": 15, "lumpi": 15, "manta": 15, "delli": 15, "find": 15, "were": 15, "least": 15, "caller": 15, "annot": 15, "venn": 15, "histogram": 15, "html": 15, "file": 15, "summari": 16, "stat": 16, "tabl": 16, "us": 16, "r": 16, "tableon": 16, "packag": 16, "telomer": 17, "length": 17, "match": 17, "sequenc": 17, "background": 17, "telomerehunt": 17, "telseq": 17, "exampl": 17, "sa": 18, "script": 18, "spider": 18, "i": 18, "prepar": 18, "epidemiologi": 18, "screen": 18, "exam": 18, "usc": 18, "appli": 18, "predict": 18, "theori": 18}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinx.ext.viewcode": 1, "sphinx": 57}, "alltitles": {"CNV calling pipeline": [[0, "cnv-calling-pipeline"]], "GATK somatic CNV calling pipeline:": [[0, "gatk-somatic-cnv-calling-pipeline"]], "CNVkit pipeline": [[0, "cnvkit-pipeline"]], "Preprocessing to create Theta inputs": [[0, "preprocessing-to-create-theta-inputs"]], "Gistic pipeline": [[0, "gistic-pipeline"]], "Chord diagram": [[1, "chord-diagram"]], "Meta analysis forest plot": [[3, "meta-analysis-forest-plot"]], "Correlation heatmap": [[2, "correlation-heatmap"]], "Generate funnel plot for meta-analysis": [[4, "generate-funnel-plot-for-meta-analysis"]], "GATK4 germline short variant discovery (SNPs + Indels)": [[8, "gatk4-germline-short-variant-discovery-snps-indels"]], "Data pre-processing for variant discovery GATK4": [[7, "data-pre-processing-for-variant-discovery-gatk4"]], "The main steps in the preprocessing workflow. A figure from GITC.": [[7, "the-main-steps-in-the-preprocessing-workflow-a-figure-from-gitc"]], "Step 0: QC": [[7, "step-0-qc"]], "Step 1: Fastqs to Unmapped BAMs": [[7, "step-1-fastqs-to-unmapped-bams"]], "Step 2: Unmapped BAMs to Mapped BAMs": [[7, "step-2-unmapped-bams-to-mapped-bams"]], "Step 3: Merge Unmapped BAMs and Mapped BAMs": [[7, "step-3-merge-unmapped-bams-and-mapped-bams"]], "Step 4: MarkDuplicates": [[7, "step-4-markduplicates"]], "Step 5: Base Quality Score Recalibration": [[7, "step-5-base-quality-score-recalibration"]], "Epigenome-wide association study": [[6, "epigenome-wide-association-study"]], "Regression coefficients and CIs": [[5, "regression-coefficients-and-cis"]], "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam @usc": [[18, "sas-scripts-and-spider-diagram-that-i-prepared-for-the-epidemiology-phd-screening-exam-usc"]], "apply exam - regression estimation and prediction": [[18, "apply-exam-regression-estimation-and-prediction"]], "theory - epidemiology": [[18, "theory-epidemiology"]], "Methylation quantitative trait loci analysis": [[13, "methylation-quantitative-trait-loci-analysis"]], "phd-pipeline": [[11, "phd-pipeline"]], "Bioinformatics pipelines": [[11, "bioinformatics-pipelines"]], "Association studies": [[11, "association-studies"]], "Data visualization": [[11, "data-visualization"]], "Other": [[11, "other"]], "Calling Structural Variations": [[15, "calling-structural-variations"]], "Smoove (lumpy)": [[15, "smoove-lumpy"]], "Manta": [[15, "manta"]], "Delly": [[15, "delly"]], "Finding variants that were called by at least 2 callers": [[15, "finding-variants-that-were-called-by-at-least-2-callers"]], "Annotate variants": [[15, "annotate-variants"]], "Create Venn diagram, histograms and html files": [[15, "create-venn-diagram-histograms-and-html-files"]], "Mutational Signatures Pipelines": [[14, "mutational-signatures-pipelines"]], "Sigprofiler": [[14, "sigprofiler"]], "pmsignature": [[14, "pmsignature"]], "deconstructSigs": [[14, "deconstructsigs"]], "Signal website": [[14, "signal-website"]], "FitMS": [[14, "fitms"]], "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)": [[17, "a-workflow-to-estimate-telomere-length-from-matched-tumor-normal-whole-genome-sequencing-data-bams"]], "Background": [[17, "background"]], "TelomereHunter": [[17, "telomerehunter"]], "TelSeq": [[17, "telseq"]], "Workflow": [[17, "workflow"]], "Example correlation plot": [[17, "example-correlation-plot"]], "GATK4 mitochondria SNPs and INDELs pipeline": [[9, "gatk4-mitochondria-snps-and-indels-pipeline"]], "Step 0: Download GATK\u2019s public reference databases": [[9, "step-0-download-gatk-s-public-reference-databases"]], "Step 1: Subset a whole genome bam to just Mitochondria reads and remove alignment information": [[9, "step-1-subset-a-whole-genome-bam-to-just-mitochondria-reads-and-remove-alignment-information"]], "Step 2: Align unmapped bam, mark duplicates, collect coverage information, and call SNPs and INDELs from normal bams": [[9, "step-2-align-unmapped-bam-mark-duplicates-collect-coverage-information-and-call-snps-and-indels-from-normal-bams"]], "Step 3: Call SNPs and INDELs from tumor bams": [[9, "step-3-call-snps-and-indels-from-tumor-bams"]], "Step 4: Estimate levels of contamination in mitochondria": [[9, "step-4-estimate-levels-of-contamination-in-mitochondria"]], "Causal mediation analysis": [[12, "causal-mediation-analysis"]], "GATK4 call somatic snvs indels": [[10, "gatk4-call-somatic-snvs-indels"]], "Best practices for somatic short variant discovery. A figure from GITC.": [[10, "best-practices-for-somatic-short-variant-discovery-a-figure-from-gitc"]], "Step 1: Scatter interval for Mutect2 parallel calling": [[10, "step-1-scatter-interval-for-mutect2-parallel-calling"]], "Step 2: Gather results from scattered intervals": [[10, "step-2-gather-results-from-scattered-intervals"]], "Step 3: Calculate contamination and learn orientation bias": [[10, "step-3-calculate-contamination-and-learn-orientation-bias"]], "Step 4: Filter alignment artifacts": [[10, "step-4-filter-alignment-artifacts"]], "Step 5: Filter variants": [[10, "step-5-filter-variants"]], "Create summary stats table using R tableone package": [[16, "create-summary-stats-table-using-r-tableone-package"]]}, "indexentries": {}})