Search.setIndex({"docnames": ["cnv/README", "gatk_data_preprocessing/README", "gatk_germline_snp_indel/README", "gatk_mitochondria_snp_indel/README", "gatk_somatic_snv_indel/README", "index", "meta_analysis_funnel_plot/README", "mutational_signatures/README", "sv/README", "tableone/README", "telomere_length/README", "year_one_screening_exam/README"], "filenames": ["cnv/README.md", "gatk_data_preprocessing/README.md", "gatk_germline_snp_indel/README.md", "gatk_mitochondria_snp_indel/README.md", "gatk_somatic_snv_indel/README.md", "index.rst", "meta_analysis_funnel_plot/README.md", "mutational_signatures/README.md", "sv/README.md", "tableone/README.md", "telomere_length/README.md", "year_one_screening_exam/README.md"], "titles": ["CNV calling pipeline", "Data pre-processing for variant discovery GATK4", "GATK4 germline short variant discovery (SNPs + Indels)", "GATK4 mitochondria SNPs and INDELs pipeline", "GATK4 call somatic snvs indels", "phd-pipeline", "Generate funnel plot for meta-analysis", "Mutational Signatures Pipelines", "Calling Structural Variations", "Create summary stats table using R tableone package", "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)", "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam &#64;usc"], "terms": {"thi": [0, 1, 3, 4, 6, 7, 8], "i": [0, 3, 4, 5, 7, 8], "copi": 0, "number": [0, 10], "variat": [0, 4], "from": [0, 2, 7, 8], "76": [0, 8], "match": [0, 1, 3, 4, 8], "tumor": [0, 1, 4, 8], "normal": [0, 1, 4, 8], "whole": [0, 1, 4, 8], "genom": [0, 1, 4, 8], "sequenc": [0, 1, 3, 4, 8], "wg": [0, 1, 2, 3, 4, 8, 10], "bam": [0, 4, 8], "38": [0, 8], "childhood": [0, 1, 3, 4, 8, 10], "acut": [0, 1, 3, 4, 8, 10], "lymphoblast": [0, 1, 3, 4, 8, 10], "leukemia": [0, 1, 3, 4, 8, 10], "case": [0, 1, 3, 4, 8, 10], "file": [0, 1, 7, 10], "have": [0, 4, 7], "been": [0, 7], "data": [0, 3, 4, 5], "check": [0, 1, 2, 3, 4], "out": [0, 1, 2, 3, 4], "best": [0, 2, 3], "practic": [0, 1, 2, 3], "document": [0, 1, 2, 4, 10], "cnv_somatic_pair_workflow": 0, "wdl": [0, 1, 2, 3, 4], "how": [0, 4], "part": 0, "sensit": 0, "detect": [0, 4, 7], "ratio": 0, "alter": [0, 4], "allel": [0, 4], "segment": 0, "ii": 0, "legaci": 0, "tutori": [0, 1], "0": [0, 4], "extractintervalfrompon": 0, "sh": [0, 1, 3, 4, 8], "1": [0, 2, 8, 10], "9": [0, 3], "funcotateseg": 0, "workflow": [0, 2, 3, 4, 5], "A": 0, "figur": 0, "gitc": 0, "2": [0, 10], "cnvkitaccess": 0, "3": [0, 8, 10], "iamp21cnvkitplot": 0, "cnvkittheta": 0, "cnvkitimporttheta": 0, "4": [0, 8, 10], "gisticcnvkitexportseg": 0, "gisticmergeseg": 0, "r": [0, 3, 7], "an": [1, 2, 3, 4, 8, 10], "hpc": [1, 2, 3, 4, 8], "50": [1, 4], "25": [1, 3, 4, 10], "were": [1, 4, 10], "illumina": [1, 10], "novaseq": [1, 10], "6000": [1, 10], "gatk": [1, 2, 4, 7, 10], "doc": [1, 3, 4], "more": [1, 2, 3, 4], "detail": [1, 2, 3, 4], "wa": [1, 4, 10], "written": [1, 4], "accord": [1, 2, 4], "follow": [1, 4], "offici": [1, 4], "version": 1, "http": [1, 2, 4, 6, 10], "github": [1, 2, 4, 10], "com": [1, 2, 4, 6, 10], "pipelin": [1, 2, 4, 8, 10], "blob": [1, 2], "master": [1, 2, 4], "task": [1, 2], "unmappedbamtoalignedbam": 1, "wholegenomegermlinesinglesampl": 1, "simpl": 1, "broadgsa": 1, "doc_arch": 1, "method": 1, "reference_implement": 1, "_pairedendsinglesamplewf_pipelin": 1, "md": 1, "sortandfixtag": 1, "other": [1, 4], "introduct": 1, "biowulf": 1, "nih": 1, "howto": 1, "run": [1, 2, 10], "bqsr": 1, "fastqc": 1, "pairedfastqtounmappedbam": 1, "validatesamfil": 1, "sortsam": 1, "bwa": 1, "sortalignedbam": 1, "validatealignedbam": 1, "checkifaltawarealign": 1, "mergebamalign": 1, "validatemergedbam": 1, "countread": 1, "validatesortedmarkedbam": 1, "baserecalibr": 1, "bqsrplot": 1, "rmd": [1, 7], "haplotypecal": 2, "gvcf": 2, "mode": [2, 4], "broad": 2, "prod": 2, "jointgenotypingwf": 2, "broadinstitut": [2, 4], "warp": 2, "jointgenotypingtask": 2, "dna_seq": 2, "joint_genotyp": 2, "jointgenotyp": 2, "mitochondri": [3, 5], "larg": 3, "base": 3, "subsetbamtochrm": 3, "revertsam": 3, "aligntomt": 3, "aligntoshiftedmt": 3, "collectwgsmetr": 3, "runcollectwgsmetricsr": 3, "callmt": 3, "germlin": [3, 4, 5], "5": [3, 7, 8], "callshiftedmt": 3, "6": [3, 4], "liftoverandcombinevcf": 3, "7": [3, 10], "mergestat": [3, 4], "8": 3, "initialfilt": 3, "splitmultiallelicsandremovenonpasssit": 3, "getcontamin": 3, "filtercontamin": 3, "mutat": [4, 5], "includ": [4, 10], "singl": 4, "nucleotid": 4, "sna": 4, "insert": 4, "delet": 4, "preprocess": 4, "tree": 4, "script": 4, "mutect2_wdl": 4, "explan": 4, "deprec": 4, "new": 4, "pdf": 4, "featur": 4, "improv": 4, "what": 4, "tradit": 4, "entail": 4, "faq": 4, "differ": [4, 7], "onli": 4, "versu": 4, "doe": 4, "do": 4, "each": [4, 10], "In": 4, "us": [4, 7, 8, 10], "popul": 4, "frequenc": 4, "resourc": 4, "prior": 4, "read": [4, 10], "fraction": 4, "\u00bd": 4, "ar": 4, "suggest": 4, "het": 4, "evid": 4, "miss": 4, "": 4, "same": 4, "model": 4, "addition": 4, "power": 4, "avail": 4, "exclud": 4, "rare": 4, "captur": 4, "individu": [4, 10], "specif": 4, "toward": 4, "being": 4, "The": 4, "simplifi": 4, "site": 4, "gnomad": 4, "retain": 4, "ftp": 4, "gsapubftp": 4, "anonym": 4, "org": 4, "bundl": 4, "panel": 4, "pon": 4, "ha": 4, "vital": 4, "role": 4, "fill": 4, "gap": 4, "between": 4, "catch": 4, "addit": 4, "nois": 4, "like": 4, "map": 4, "somewhat": 4, "random": 4, "systemat": 4, "process": [4, 5, 7, 10], "interest": 4, "biostar": 4, "post": 4, "heterozygot": 4, "And": 4, "manifest": 4, "project": [4, 5], "creat": [4, 5], "contain": 4, "artifactu": 4, "tool": 4, "take": [4, 10], "multipl": 4, "sampl": [4, 10], "callset": 4, "produc": 4, "collat": 4, "present": 4, "two": [4, 10], "default": 4, "set": 4, "min": 4, "count": 4, "argument": 4, "vcf": [4, 7], "common": 4, "level": 4, "max": 4, "probabl": 4, "threshold": [4, 10], "possibl": 4, "By": 4, "so": 4, "event": 4, "usual": 4, "correct": 4, "behavior": 4, "handl": 4, "probabilist": 4, "via": 4, "human": 4, "much": 4, "refin": 4, "than": 4, "ani": 4, "could": 4, "final": [4, 10], "tabl": [4, 5, 10], "tell": 4, "u": 4, "rate": 4, "cross": 4, "If": 4, "estim": [4, 5], "15": 4, "error": 4, "19": 4, "mean": 4, "one": [4, 5], "everi": 4, "hundr": 4, "come": 4, "someon": 4, "els": 4, "percentag": 4, "effect": 4, "our": 4, "floor": 4, "low": 4, "potenti": 4, "below": 4, "af": 4, "we": [4, 10], "would": 4, "zero": 4, "judg": 4, "whether": 4, "thei": 4, "real": 4, "through": 4, "feed": 4, "can": 4, "properli": 4, "account": 4, "scatterinterv": 4, "mutect2scattergather0": 4, "mergevcf": 4, "mergebamout": 4, "orientationbia": 4, "createbwaimag": 4, "filteralignmentartifact": 4, "funcot": 4, "bcftoolfilt": 4, "some": 5, "wrote": 5, "my": 5, "research": 5, "cnv": 5, "epigenom": 5, "wide": 5, "associ": 5, "studi": [5, 6], "pre": [5, 10], "variant": [5, 10], "discoveri": [5, 10], "gatk4": 5, "short": 5, "snp": 5, "indel": 5, "snv": 5, "call": 5, "somat": 5, "mediat": 5, "analysi": 5, "meta": 5, "funnel": 5, "plot": 5, "methyl": 5, "quantit": 5, "trait": 5, "loci": 5, "signatur": 5, "sv": 5, "summari": [5, 8], "stat": 5, "telomer": 5, "length": 5, "year": 5, "screen": 5, "exam": 5, "fig2": 6, "fig3": 6, "www": 6, "liebertpub": 6, "doi": 6, "ab": 6, "10": 6, "1089": 6, "jicm": 6, "2022": 6, "0586": 6, "mutect2": 7, "algorithm": 7, "step1": 7, "py": 7, "step2": 7, "step3": 7, "step4": 7, "signalsummari": 7, "signalkataegi": 7, "step5": 7, "three": 8, "softwar": [8, 10], "premergesvpersamplesmoov": 8, "smooveunionsit": 8, "smoovegenotyp": 8, "smoovepasteannot": 8, "premergesvpersamplemantaconfigur": 8, "mantaworkflow": 8, "mantaclean": 8, "dellypersamplecal": 8, "dellyprefilt": 8, "dellygeno": 8, "dellypostfilt": 8, "dellybcf2vcf": 8, "survivor": 8, "anno": 8, "exampl": 8, "graph": 8, "appli": 10, "gener": 10, "compar": 10, "result": 10, "paper": 10, "content": 10, "quantifi": 10, "ten": 10, "repeat": 10, "tcaggg": 10, "tgaggg": 10, "ttgggg": 10, "ttcggg": 10, "tttggg": 10, "ataggg": 10, "catggg": 10, "ctaggg": 10, "gtaggg": 10, "taaggg": 10, "tl": 10, "wsa": 10, "kb": 10, "ttaggg": 10, "ccctaa": 10, "consid": 10, "step": 10, "aggreg": 10, "output": 10, "clean": 10, "calcul": 10, "need": 10, "weight": 10, "averag": 10, "all": 10, "group": 10, "within": 10, "zd1": 10, "issu": 10, "had": 10, "high": 10, "code": 11}, "objects": {}, "objtypes": {}, "objnames": {}, "titleterms": {"cnv": 0, "call": [0, 3, 4, 8], "pipelin": [0, 3, 5, 7], "gatk": [0, 3], "somat": [0, 4], "cnvkit": 0, "preprocess": [0, 1], "creat": [0, 8, 9], "theta": 0, "input": 0, "gistic": 0, "data": [1, 10], "pre": 1, "process": 1, "variant": [1, 2, 4, 8], "discoveri": [1, 2, 4], "gatk4": [1, 2, 3, 4], "The": 1, "main": 1, "step": [1, 3, 4], "workflow": [1, 10], "A": [1, 4, 10], "figur": [1, 4], "from": [1, 3, 4, 10], "gitc": [1, 4], "0": [1, 3], "qc": 1, "1": [1, 3, 4], "fastq": 1, "unmap": [1, 3], "bam": [1, 3, 10], "2": [1, 3, 4, 8], "map": 1, "3": [1, 3, 4], "merg": 1, "4": [1, 3, 4], "markdupl": 1, "5": [1, 4], "base": 1, "qualiti": 1, "score": 1, "recalibr": 1, "germlin": 2, "short": [2, 4], "snp": [2, 3], "indel": [2, 3, 4], "mitochondria": 3, "download": 3, "": 3, "public": 3, "refer": 3, "databas": 3, "subset": 3, "whole": [3, 10], "genom": [3, 10], "just": 3, "read": 3, "remov": 3, "align": [3, 4], "inform": 3, "mark": 3, "duplic": 3, "collect": 3, "coverag": 3, "normal": [3, 10], "tumor": [3, 10], "estim": [3, 10, 11], "level": 3, "contamin": [3, 4], "snv": 4, "best": 4, "practic": 4, "scatter": 4, "interv": 4, "mutect2": 4, "parallel": 4, "gather": 4, "result": 4, "calcul": 4, "learn": 4, "orient": 4, "bia": 4, "filter": 4, "artifact": 4, "phd": [5, 11], "gener": 6, "funnel": 6, "plot": [6, 10], "meta": 6, "analysi": 6, "mutat": 7, "signatur": 7, "sigprofil": 7, "pmsignatur": 7, "deconstructsig": 7, "signal": 7, "websit": 7, "fitm": 7, "structur": 8, "variat": 8, "smoov": 8, "lumpi": 8, "manta": 8, "delli": 8, "find": 8, "were": 8, "least": 8, "caller": 8, "annot": 8, "venn": 8, "diagram": [8, 11], "histogram": 8, "html": 8, "file": 8, "summari": 9, "stat": 9, "tabl": 9, "us": 9, "r": 9, "tableon": 9, "packag": 9, "telomer": 10, "length": 10, "match": 10, "sequenc": 10, "background": 10, "telomerehunt": 10, "telseq": 10, "exampl": 10, "correl": 10, "sa": 11, "script": 11, "spider": 11, "i": 11, "prepar": 11, "epidemiologi": 11, "screen": 11, "exam": 11, "usc": 11, "appli": 11, "regress": 11, "predict": 11, "theori": 11}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinx.ext.viewcode": 1, "sphinx": 57}, "alltitles": {"Mutational Signatures Pipelines": [[7, "mutational-signatures-pipelines"]], "Sigprofiler": [[7, "sigprofiler"]], "pmsignature": [[7, "pmsignature"]], "deconstructSigs": [[7, "deconstructsigs"]], "Signal website": [[7, "signal-website"]], "FitMS": [[7, "fitms"]], "Calling Structural Variations": [[8, "calling-structural-variations"]], "Smoove (lumpy)": [[8, "smoove-lumpy"]], "Manta": [[8, "manta"]], "Delly": [[8, "delly"]], "Finding variants that were called by at least 2 callers": [[8, "finding-variants-that-were-called-by-at-least-2-callers"]], "Annotate variants": [[8, "annotate-variants"]], "Create Venn diagram, histograms and html files": [[8, "create-venn-diagram-histograms-and-html-files"]], "Generate funnel plot for meta-analysis": [[6, "generate-funnel-plot-for-meta-analysis"]], "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)": [[10, "a-workflow-to-estimate-telomere-length-from-matched-tumor-normal-whole-genome-sequencing-data-bams"]], "Background": [[10, "background"]], "TelomereHunter": [[10, "telomerehunter"]], "TelSeq": [[10, "telseq"]], "Workflow": [[10, "workflow"]], "Example correlation plot": [[10, "example-correlation-plot"]], "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam @usc": [[11, "sas-scripts-and-spider-diagram-that-i-prepared-for-the-epidemiology-phd-screening-exam-usc"]], "apply exam - regression estimation and prediction": [[11, "apply-exam-regression-estimation-and-prediction"]], "theory - epidemiology": [[11, "theory-epidemiology"]], "Create summary stats table using R tableone package": [[9, "create-summary-stats-table-using-r-tableone-package"]], "CNV calling pipeline": [[0, "cnv-calling-pipeline"]], "GATK somatic CNV calling pipeline:": [[0, "gatk-somatic-cnv-calling-pipeline"]], "CNVkit pipeline": [[0, "cnvkit-pipeline"]], "Preprocessing to create Theta inputs": [[0, "preprocessing-to-create-theta-inputs"]], "Gistic pipeline": [[0, "gistic-pipeline"]], "GATK4 germline short variant discovery (SNPs + Indels)": [[2, "gatk4-germline-short-variant-discovery-snps-indels"]], "Data pre-processing for variant discovery GATK4": [[1, "data-pre-processing-for-variant-discovery-gatk4"]], "The main steps in the preprocessing workflow. A figure from GITC.": [[1, "the-main-steps-in-the-preprocessing-workflow-a-figure-from-gitc"]], "Step 0: QC": [[1, "step-0-qc"]], "Step 1: Fastqs to Unmapped BAMs": [[1, "step-1-fastqs-to-unmapped-bams"]], "Step 2: Unmapped BAMs to Mapped BAMs": [[1, "step-2-unmapped-bams-to-mapped-bams"]], "Step 3: Merge Unmapped BAMs and Mapped BAMs": [[1, "step-3-merge-unmapped-bams-and-mapped-bams"]], "Step 4: MarkDuplicates": [[1, "step-4-markduplicates"]], "Step 5: Base Quality Score Recalibration": [[1, "step-5-base-quality-score-recalibration"]], "GATK4 call somatic snvs indels": [[4, "gatk4-call-somatic-snvs-indels"]], "Best practices for somatic short variant discovery. A figure from GITC.": [[4, "best-practices-for-somatic-short-variant-discovery-a-figure-from-gitc"]], "Step 1: Scatter interval for Mutect2 parallel calling": [[4, "step-1-scatter-interval-for-mutect2-parallel-calling"]], "Step 2: Gather results from scattered intervals": [[4, "step-2-gather-results-from-scattered-intervals"]], "Step 3: Calculate contamination and learn orientation bias": [[4, "step-3-calculate-contamination-and-learn-orientation-bias"]], "Step 4: Filter alignment artifacts": [[4, "step-4-filter-alignment-artifacts"]], "Step 5: Filter variants": [[4, "step-5-filter-variants"]], "GATK4 mitochondria SNPs and INDELs pipeline": [[3, "gatk4-mitochondria-snps-and-indels-pipeline"]], "Step 0: Download GATK\u2019s public reference databases": [[3, "step-0-download-gatk-s-public-reference-databases"]], "Step 1: Subset a whole genome bam to just Mitochondria reads and remove alignment information": [[3, "step-1-subset-a-whole-genome-bam-to-just-mitochondria-reads-and-remove-alignment-information"]], "Step 2: Align unmapped bam, mark duplicates, collect coverage information, and call SNPs and INDELs from normal bams": [[3, "step-2-align-unmapped-bam-mark-duplicates-collect-coverage-information-and-call-snps-and-indels-from-normal-bams"]], "Step 3: Call SNPs and INDELs from tumor bams": [[3, "step-3-call-snps-and-indels-from-tumor-bams"]], "Step 4: Estimate levels of contamination in mitochondria": [[3, "step-4-estimate-levels-of-contamination-in-mitochondria"]], "phd-pipeline": [[5, "phd-pipeline"]]}, "indexentries": {}})