Search.setIndex({"docnames": ["cnv/README", "data_visualization/chord_diagram/README", "data_visualization/heatmap/README", "data_visualization/meta_analysis_forest_plot/README", "data_visualization/regression_coefficient_ci/README", "ewas/README", "gatk_data_preprocessing/README", "gatk_germline_snp_indel/README", "gatk_mitochondria_snp_indel/README", "gatk_somatic_snv_indel/README", "index", "mediation_analysis/README", "meta_analysis_funnel_plot/README", "mqtl/README", "mutational_signatures/README", "sv/README", "tableone/README", "telomere_length/README", "year_one_screening_exam/README"], "filenames": ["cnv/README.md", "data_visualization/chord_diagram/README.md", "data_visualization/heatmap/README.md", "data_visualization/meta_analysis_forest_plot/README.md", "data_visualization/regression_coefficient_ci/README.md", "ewas/README.md", "gatk_data_preprocessing/README.md", "gatk_germline_snp_indel/README.md", "gatk_mitochondria_snp_indel/README.md", "gatk_somatic_snv_indel/README.md", "index.rst", "mediation_analysis/README.md", "meta_analysis_funnel_plot/README.md", "mqtl/README.md", "mutational_signatures/README.md", "sv/README.md", "tableone/README.md", "telomere_length/README.md", "year_one_screening_exam/README.md"], "titles": ["CNV calling pipeline", "Chord diagram", "Correlation heatmap", "Meta analysis forest plot", "Regression coefficients and CIs", "Epigenome-wide association study", "Data pre-processing for variant discovery GATK4", "GATK4 germline short variant discovery (SNPs + Indels)", "GATK4 mitochondria SNPs and INDELs pipeline", "GATK4 call somatic snvs indels", "phd-pipeline", "Causal mediation analysis", "Generate funnel plot for meta-analysis", "Methylation quantitative trait loci analysis", "Mutational Signatures Pipelines", "Calling Structural Variations", "Create summary stats table using R tableone package", "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)", "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam &#64;usc"], "terms": {"thi": [0, 1, 2, 3, 4, 6, 8, 9, 11, 12, 13, 14, 15], "i": [0, 1, 2, 3, 4, 8, 9, 10, 11, 13, 14, 15], "copi": 0, "number": [0, 1, 13, 17], "variat": [0, 9], "from": [0, 4, 5, 7, 14, 15], "76": [0, 15], "match": [0, 6, 8, 9, 15], "tumor": [0, 1, 6, 9, 15], "normal": [0, 6, 9, 11, 15], "whole": [0, 6, 9, 15], "genom": [0, 6, 9, 10, 15], "sequenc": [0, 6, 8, 9, 15], "wg": [0, 6, 7, 8, 9, 15, 17], "bam": [0, 9, 15], "38": [0, 15], "childhood": [0, 6, 8, 9, 13, 15, 17], "acut": [0, 6, 8, 9, 15, 17], "lymphoblast": [0, 6, 8, 9, 15, 17], "leukemia": [0, 6, 8, 9, 15, 17], "case": [0, 5, 6, 8, 9, 15, 17], "file": [0, 6, 14, 17], "have": [0, 9, 14], "been": [0, 14], "data": [0, 8, 9], "check": [0, 6, 7, 8, 9], "out": [0, 6, 7, 8, 9], "best": [0, 7, 8], "practic": [0, 6, 7, 8], "document": [0, 6, 7, 9, 17], "cnv_somatic_pair_workflow": 0, "wdl": [0, 6, 7, 8, 9], "how": [0, 9], "part": 0, "sensit": 0, "detect": [0, 9, 14], "ratio": 0, "alter": [0, 9], "allel": [0, 9], "segment": 0, "ii": [0, 11], "legaci": 0, "tutori": [0, 6], "0": [0, 9, 13], "extractintervalfrompon": 0, "sh": [0, 6, 8, 9, 15], "1": [0, 7, 13, 15, 17], "9": [0, 8], "funcotateseg": 0, "workflow": [0, 5, 7, 8, 9, 10, 11, 13], "A": [0, 5], "figur": 0, "gitc": 0, "2": [0, 13, 17], "cnvkitaccess": 0, "3": [0, 15, 17], "iamp21cnvkitplot": 0, "cnvkittheta": 0, "cnvkitimporttheta": 0, "4": [0, 15, 17], "gisticcnvkitexportseg": 0, "gisticmergeseg": 0, "r": [0, 5, 8, 11, 13, 14], "script": [1, 2, 3, 4, 5, 9, 11, 13, 16], "gener": [1, 5, 16, 17], "show": [1, 2, 4, 11], "gene": 1, "delet": [1, 9], "frequenc": [1, 9], "eight": 1, "commonli": 1, "region": 1, "cdkn2a": 1, "etv6": 1, "ikzf1": 1, "pax5": 1, "btg1": 1, "ebf1": 1, "rb1": 1, "within": [1, 17], "pseudoautosom": 1, "par1": 1, "sex": [1, 5, 11], "chromosom": [1, 5], "crlf2": 1, "csf2ra": 1, "il3ra": 1, "thei": [1, 9], "were": [1, 5, 6, 9, 11, 13, 17], "assai": 1, "dna": [1, 5, 13], "us": [1, 5, 9, 11, 13, 14, 15, 17], "multiplex": 1, "ligat": 1, "depend": [1, 13], "probe": 1, "amplif": 1, "mlpa": 1, "studi": [1, 2, 3, 4, 11, 12, 13, 16], "public": [1, 2, 3, 4, 5, 11, 13], "link": [1, 2, 3, 4, 5, 11, 13, 16], "output": [1, 2, 3, 4, 17], "creat": [2, 3, 4, 9, 10], "pearson": 2, "coeffici": [2, 10], "self": 2, "report": 2, "smoke": 2, "variabl": 2, "plot": [4, 5, 10], "multipl": [4, 9, 13], "run": [5, 6, 7, 17], "ewa": 5, "analys": [5, 11], "identifi": [5, 13], "all": [5, 11, 13, 17], "differenti": 5, "methyl": [5, 10, 11], "cpg": [5, 13], "posit": 5, "autosom": 5, "we": [5, 9, 11, 13, 17], "fit": [5, 11, 13], "logist": 5, "regress": [5, 10, 13], "model": [5, 9, 11, 13], "predict": [5, 13], "control": 5, "statu": [5, 11], "function": [5, 11, 13], "each": [5, 9, 13, 17], "remain": 5, "adjust": [5, 13], "batch": [5, 11], "effect": [5, 9, 11, 13], "cell": [5, 11], "type": [5, 11], "heterogen": [5, 11], "first": [5, 11, 13], "10": [5, 12], "princip": 5, "compon": 5, "deriv": 5, "refactor": 5, "58": 5, "genet": 5, "ancestri": [5, 11], "epistructur": 5, "fix": [5, 11, 13], "meta": [5, 10, 11, 13], "analysi": [5, 10], "summari": [5, 10, 15], "estim": [5, 9, 10, 11], "result": [5, 11, 13, 17], "overlap": 5, "hree": 5, "differ": [5, 9, 14], "set": [5, 9], "packag": [5, 11, 13], "metafor": [5, 13], "bidirect": 5, "manhattan": 5, "qq": 5, "sampl": [5, 9, 16, 17], "cmplot": 5, "rmd": [5, 6, 14], "an": [6, 7, 8, 9, 11, 13, 15, 17], "hpc": [6, 7, 8, 9, 15], "50": [6, 9], "25": [6, 8, 9, 17], "illumina": [6, 17], "novaseq": [6, 17], "6000": [6, 17], "gatk": [6, 7, 9, 14, 17], "doc": [6, 8, 9], "more": [6, 7, 8, 9], "detail": [6, 7, 8, 9], "wa": [6, 9, 11, 13, 17], "written": [6, 9], "accord": [6, 7, 9], "follow": [6, 9], "offici": [6, 9], "version": 6, "http": [6, 7, 9, 12, 17], "github": [6, 7, 9, 17], "com": [6, 7, 9, 12, 17], "pipelin": [6, 7, 9, 15, 17], "blob": [6, 7], "master": [6, 7, 9], "task": [6, 7], "unmappedbamtoalignedbam": 6, "wholegenomegermlinesinglesampl": 6, "simpl": 6, "broadgsa": 6, "doc_arch": 6, "method": 6, "reference_implement": 6, "_pairedendsinglesamplewf_pipelin": 6, "md": 6, "sortandfixtag": 6, "other": [6, 9], "introduct": 6, "biowulf": 6, "nih": 6, "howto": 6, "bqsr": 6, "fastqc": 6, "pairedfastqtounmappedbam": 6, "validatesamfil": 6, "sortsam": 6, "bwa": 6, "sortalignedbam": 6, "validatealignedbam": 6, "checkifaltawarealign": 6, "mergebamalign": 6, "validatemergedbam": 6, "countread": 6, "validatesortedmarkedbam": 6, "baserecalibr": 6, "bqsrplot": 6, "haplotypecal": 7, "gvcf": 7, "mode": [7, 9], "broad": 7, "prod": 7, "jointgenotypingwf": 7, "broadinstitut": [7, 9], "warp": 7, "jointgenotypingtask": 7, "dna_seq": 7, "joint_genotyp": 7, "jointgenotyp": 7, "mitochondri": [8, 10], "larg": 8, "base": [8, 11], "subsetbamtochrm": 8, "revertsam": 8, "aligntomt": 8, "aligntoshiftedmt": 8, "collectwgsmetr": 8, "runcollectwgsmetricsr": 8, "callmt": 8, "germlin": [8, 9, 10], "5": [8, 14, 15], "callshiftedmt": 8, "6": [8, 9], "liftoverandcombinevcf": 8, "7": [8, 17], "mergestat": [8, 9], "8": 8, "initialfilt": 8, "splitmultiallelicsandremovenonpasssit": 8, "getcontamin": 8, "filtercontamin": 8, "mutat": [9, 10], "includ": [9, 11, 17], "singl": 9, "nucleotid": 9, "sna": 9, "insert": 9, "preprocess": 9, "tree": 9, "mutect2_wdl": 9, "explan": 9, "deprec": 9, "new": 9, "pdf": 9, "featur": 9, "improv": 9, "what": 9, "tradit": 9, "entail": 9, "faq": 9, "onli": 9, "versu": 9, "doe": 9, "do": 9, "In": 9, "popul": 9, "resourc": 9, "prior": 9, "read": [9, 17], "fraction": 9, "\u00bd": 9, "ar": 9, "suggest": 9, "het": 9, "evid": 9, "miss": 9, "": 9, "same": [9, 11], "addition": 9, "power": 9, "avail": 9, "exclud": 9, "rare": 9, "captur": 9, "individu": [9, 17], "specif": 9, "toward": 9, "being": 9, "The": [9, 11, 13], "simplifi": 9, "site": [9, 13], "gnomad": 9, "retain": 9, "ftp": 9, "gsapubftp": 9, "anonym": 9, "org": 9, "bundl": 9, "panel": 9, "pon": 9, "ha": 9, "vital": 9, "role": 9, "fill": 9, "gap": 9, "between": [9, 13], "catch": 9, "addit": [9, 13], "nois": 9, "like": 9, "map": 9, "somewhat": 9, "random": 9, "systemat": 9, "process": [9, 10, 14, 17], "interest": 9, "biostar": 9, "post": 9, "heterozygot": 9, "And": 9, "manifest": 9, "project": [9, 10], "contain": 9, "artifactu": 9, "tool": 9, "take": [9, 17], "callset": 9, "produc": 9, "collat": 9, "present": 9, "two": [9, 11, 17], "default": 9, "min": 9, "count": 9, "argument": 9, "vcf": [9, 14], "common": 9, "level": [9, 11], "max": 9, "probabl": 9, "threshold": [9, 13, 17], "possibl": 9, "By": 9, "so": 9, "event": 9, "usual": 9, "correct": [9, 13], "behavior": 9, "handl": 9, "probabilist": 9, "via": 9, "human": 9, "much": 9, "refin": 9, "than": 9, "ani": 9, "could": 9, "final": [9, 17], "tabl": [9, 10, 17], "tell": 9, "u": 9, "rate": 9, "cross": 9, "If": 9, "15": 9, "error": 9, "19": 9, "mean": 9, "one": [9, 10, 16], "everi": 9, "hundr": 9, "come": 9, "someon": 9, "els": 9, "percentag": 9, "our": 9, "floor": 9, "low": 9, "potenti": 9, "below": 9, "af": 9, "would": 9, "zero": 9, "judg": 9, "whether": 9, "real": 9, "through": 9, "feed": 9, "can": 9, "properli": 9, "account": 9, "scatterinterv": 9, "mutect2scattergather0": 9, "mergevcf": 9, "mergebamout": 9, "orientationbia": 9, "createbwaimag": 9, "filteralignmentartifact": 9, "funcot": 9, "bcftoolfilt": 9, "some": 10, "wrote": 10, "my": 10, "research": 10, "pre": [10, 17], "variant": [10, 17], "discoveri": [10, 17], "gatk4": 10, "short": 10, "snp": [10, 13], "indel": 10, "call": 10, "somat": 10, "snv": 10, "sv": 10, "cnv": 10, "signatur": 10, "telomer": 10, "length": 10, "eigenet": 10, "clock": 10, "epigenom": 10, "wide": 10, "quantit": 10, "trait": 10, "loci": 10, "causal": 10, "mediat": 10, "stat": 10, "chord": 10, "diagram": [10, 11], "correl": 10, "heatmap": 10, "ci": 10, "funnel": 10, "forest": 10, "year": 10, "screen": 10, "exam": 10, "perform": 11, "signific": 11, "mqtl": [11, 13], "dmp": [11, 13], "pair": 11, "specifi": 11, "statist": 11, "distribut": 11, "after": 11, "condit": 11, "genotyp": [11, 13], "covari": 11, "outcom": 11, "given": 11, "separ": [11, 13], "paramet": 11, "main": 11, "input": 11, "which": 11, "comput": 11, "acm": 11, "ad": 11, "total": 11, "varianc": 11, "simul": 11, "quasi": 11, "bayesian": 11, "mont": 11, "carlo": 11, "approxim": 11, "conduct": [11, 13], "1000": 11, "time": 11, "altern": 11, "approach": 11, "non": 11, "parametr": 11, "bootstrap": 11, "also": 11, "appli": [11, 17], "valid": 11, "three": [11, 13, 15], "dataset": [11, 13], "summar": 11, "across": [11, 13], "path": 11, "confound": 11, "fig2": 12, "fig3": 12, "www": 12, "liebertpub": 12, "doi": 12, "ab": 12, "1089": 12, "jicm": 12, "2022": 12, "0586": 12, "associ": 13, "risk": 13, "matrix": 13, "eqtl": 13, "linear": 13, "code": [13, 18], "test": 13, "stringent": 13, "bonferroni": 13, "025": 13, "fdr": 13, "05": 13, "subsequ": 13, "analyz": 13, "mutect2": 14, "algorithm": 14, "step1": 14, "py": 14, "step2": 14, "step3": 14, "step4": 14, "signalsummari": 14, "signalkataegi": 14, "step5": 14, "softwar": [15, 17], "premergesvpersamplesmoov": 15, "smooveunionsit": 15, "smoovegenotyp": 15, "smoovepasteannot": 15, "premergesvpersamplemantaconfigur": 15, "mantaworkflow": 15, "mantaclean": 15, "dellypersamplecal": 15, "dellyprefilt": 15, "dellygeno": 15, "dellypostfilt": 15, "dellybcf2vcf": 15, "survivor": 15, "anno": 15, "exampl": 15, "graph": 15, "inform": 16, "compar": 17, "paper": 17, "content": 17, "quantifi": 17, "ten": 17, "repeat": 17, "tcaggg": 17, "tgaggg": 17, "ttgggg": 17, "ttcggg": 17, "tttggg": 17, "ataggg": 17, "catggg": 17, "ctaggg": 17, "gtaggg": 17, "taaggg": 17, "tl": 17, "wsa": 17, "kb": 17, "ttaggg": 17, "ccctaa": 17, "consid": 17, "step": 17, "aggreg": 17, "clean": 17, "calcul": 17, "need": 17, "weight": 17, "averag": 17, "group": 17, "zd1": 17, "issu": 17, "had": 17, "high": 17}, "objects": {}, "objtypes": {}, "objnames": {}, "titleterms": {"cnv": 0, "call": [0, 8, 9, 15], "pipelin": [0, 8, 10, 14], "gatk": [0, 8], "somat": [0, 9], "cnvkit": 0, "preprocess": [0, 6], "creat": [0, 15, 16], "theta": 0, "input": 0, "gistic": 0, "chord": 1, "diagram": [1, 15, 18], "correl": [2, 17], "heatmap": 2, "meta": [3, 12], "analysi": [3, 11, 12, 13], "forest": 3, "plot": [3, 12, 17], "regress": [4, 18], "coeffici": 4, "ci": 4, "epigenom": 5, "wide": 5, "associ": [5, 10], "studi": [5, 10], "data": [6, 10, 17], "pre": 6, "process": 6, "variant": [6, 7, 9, 15], "discoveri": [6, 7, 9], "gatk4": [6, 7, 8, 9], "The": 6, "main": 6, "step": [6, 8, 9], "workflow": [6, 17], "A": [6, 9, 17], "figur": [6, 9], "from": [6, 8, 9, 17], "gitc": [6, 9], "0": [6, 8], "qc": 6, "1": [6, 8, 9], "fastq": 6, "unmap": [6, 8], "bam": [6, 8, 17], "2": [6, 8, 9, 15], "map": 6, "3": [6, 8, 9], "merg": 6, "4": [6, 8, 9], "markdupl": 6, "5": [6, 9], "base": 6, "qualiti": 6, "score": 6, "recalibr": 6, "germlin": 7, "short": [7, 9], "snp": [7, 8], "indel": [7, 8, 9], "mitochondria": 8, "download": 8, "": 8, "public": 8, "refer": 8, "databas": 8, "subset": 8, "whole": [8, 17], "genom": [8, 17], "just": 8, "read": 8, "remov": 8, "align": [8, 9], "inform": 8, "mark": 8, "duplic": 8, "collect": 8, "coverag": 8, "normal": [8, 17], "tumor": [8, 17], "estim": [8, 17, 18], "level": 8, "contamin": [8, 9], "snv": 9, "best": 9, "practic": 9, "scatter": 9, "interv": 9, "mutect2": 9, "parallel": 9, "gather": 9, "result": 9, "calcul": 9, "learn": 9, "orient": 9, "bia": 9, "filter": 9, "artifact": 9, "phd": [10, 18], "bioinformat": 10, "visual": 10, "other": 10, "causal": 11, "mediat": 11, "gener": 12, "funnel": 12, "methyl": 13, "quantit": 13, "trait": 13, "loci": 13, "mutat": 14, "signatur": 14, "sigprofil": 14, "pmsignatur": 14, "deconstructsig": 14, "signal": 14, "websit": 14, "fitm": 14, "structur": 15, "variat": 15, "smoov": 15, "lumpi": 15, "manta": 15, "delli": 15, "find": 15, "were": 15, "least": 15, "caller": 15, "annot": 15, "venn": 15, "histogram": 15, "html": 15, "file": 15, "summari": 16, "stat": 16, "tabl": 16, "us": 16, "r": 16, "tableon": 16, "packag": 16, "telomer": 17, "length": 17, "match": 17, "sequenc": 17, "background": 17, "telomerehunt": 17, "telseq": 17, "exampl": 17, "sa": 18, "script": 18, "spider": 18, "i": 18, "prepar": 18, "epidemiologi": 18, "screen": 18, "exam": 18, "usc": 18, "appli": 18, "predict": 18, "theori": 18}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinx.ext.viewcode": 1, "sphinx": 57}, "alltitles": {"Chord diagram": [[1, "chord-diagram"]], "Data pre-processing for variant discovery GATK4": [[6, "data-pre-processing-for-variant-discovery-gatk4"]], "The main steps in the preprocessing workflow. A figure from GITC.": [[6, "the-main-steps-in-the-preprocessing-workflow-a-figure-from-gitc"]], "Step 0: QC": [[6, "step-0-qc"]], "Step 1: Fastqs to Unmapped BAMs": [[6, "step-1-fastqs-to-unmapped-bams"]], "Step 2: Unmapped BAMs to Mapped BAMs": [[6, "step-2-unmapped-bams-to-mapped-bams"]], "Step 3: Merge Unmapped BAMs and Mapped BAMs": [[6, "step-3-merge-unmapped-bams-and-mapped-bams"]], "Step 4: MarkDuplicates": [[6, "step-4-markduplicates"]], "Step 5: Base Quality Score Recalibration": [[6, "step-5-base-quality-score-recalibration"]], "Epigenome-wide association study": [[5, "epigenome-wide-association-study"]], "GATK4 germline short variant discovery (SNPs + Indels)": [[7, "gatk4-germline-short-variant-discovery-snps-indels"]], "Regression coefficients and CIs": [[4, "regression-coefficients-and-cis"]], "GATK4 mitochondria SNPs and INDELs pipeline": [[8, "gatk4-mitochondria-snps-and-indels-pipeline"]], "Step 0: Download GATK\u2019s public reference databases": [[8, "step-0-download-gatk-s-public-reference-databases"]], "Step 1: Subset a whole genome bam to just Mitochondria reads and remove alignment information": [[8, "step-1-subset-a-whole-genome-bam-to-just-mitochondria-reads-and-remove-alignment-information"]], "Step 2: Align unmapped bam, mark duplicates, collect coverage information, and call SNPs and INDELs from normal bams": [[8, "step-2-align-unmapped-bam-mark-duplicates-collect-coverage-information-and-call-snps-and-indels-from-normal-bams"]], "Step 3: Call SNPs and INDELs from tumor bams": [[8, "step-3-call-snps-and-indels-from-tumor-bams"]], "Step 4: Estimate levels of contamination in mitochondria": [[8, "step-4-estimate-levels-of-contamination-in-mitochondria"]], "Meta analysis forest plot": [[3, "meta-analysis-forest-plot"]], "Correlation heatmap": [[2, "correlation-heatmap"]], "CNV calling pipeline": [[0, "cnv-calling-pipeline"]], "GATK somatic CNV calling pipeline:": [[0, "gatk-somatic-cnv-calling-pipeline"]], "CNVkit pipeline": [[0, "cnvkit-pipeline"]], "Preprocessing to create Theta inputs": [[0, "preprocessing-to-create-theta-inputs"]], "Gistic pipeline": [[0, "gistic-pipeline"]], "Mutational Signatures Pipelines": [[14, "mutational-signatures-pipelines"]], "Sigprofiler": [[14, "sigprofiler"]], "pmsignature": [[14, "pmsignature"]], "deconstructSigs": [[14, "deconstructsigs"]], "Signal website": [[14, "signal-website"]], "FitMS": [[14, "fitms"]], "Create summary stats table using R tableone package": [[16, "create-summary-stats-table-using-r-tableone-package"]], "Methylation quantitative trait loci analysis": [[13, "methylation-quantitative-trait-loci-analysis"]], "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)": [[17, "a-workflow-to-estimate-telomere-length-from-matched-tumor-normal-whole-genome-sequencing-data-bams"]], "Background": [[17, "background"]], "TelomereHunter": [[17, "telomerehunter"]], "TelSeq": [[17, "telseq"]], "Workflow": [[17, "workflow"]], "Example correlation plot": [[17, "example-correlation-plot"]], "Calling Structural Variations": [[15, "calling-structural-variations"]], "Smoove (lumpy)": [[15, "smoove-lumpy"]], "Manta": [[15, "manta"]], "Delly": [[15, "delly"]], "Finding variants that were called by at least 2 callers": [[15, "finding-variants-that-were-called-by-at-least-2-callers"]], "Annotate variants": [[15, "annotate-variants"]], "Create Venn diagram, histograms and html files": [[15, "create-venn-diagram-histograms-and-html-files"]], "Causal mediation analysis": [[11, "causal-mediation-analysis"]], "Generate funnel plot for meta-analysis": [[12, "generate-funnel-plot-for-meta-analysis"]], "phd-pipeline": [[10, "phd-pipeline"]], "Bioinformatics pipelines": [[10, "bioinformatics-pipelines"]], "Association studies": [[10, "association-studies"]], "Data visualization": [[10, "data-visualization"]], "Other": [[10, "other"]], "GATK4 call somatic snvs indels": [[9, "gatk4-call-somatic-snvs-indels"]], "Best practices for somatic short variant discovery. A figure from GITC.": [[9, "best-practices-for-somatic-short-variant-discovery-a-figure-from-gitc"]], "Step 1: Scatter interval for Mutect2 parallel calling": [[9, "step-1-scatter-interval-for-mutect2-parallel-calling"]], "Step 2: Gather results from scattered intervals": [[9, "step-2-gather-results-from-scattered-intervals"]], "Step 3: Calculate contamination and learn orientation bias": [[9, "step-3-calculate-contamination-and-learn-orientation-bias"]], "Step 4: Filter alignment artifacts": [[9, "step-4-filter-alignment-artifacts"]], "Step 5: Filter variants": [[9, "step-5-filter-variants"]], "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam @usc": [[18, "sas-scripts-and-spider-diagram-that-i-prepared-for-the-epidemiology-phd-screening-exam-usc"]], "apply exam - regression estimation and prediction": [[18, "apply-exam-regression-estimation-and-prediction"]], "theory - epidemiology": [[18, "theory-epidemiology"]]}, "indexentries": {}})