Search.setIndex({"docnames": ["cnv/README", "ewas/README", "gatk_data_preprocessing/README", "gatk_germline_snp_indel/README", "gatk_mitochondria_snp_indel/README", "gatk_somatic_snv_indel/README", "index", "mediation_analysis/README", "meta_analysis_funnel_plot/README", "mqtl/README", "mutational_signatures/README", "sv/README", "tableone/README", "telomere_length/README", "year_one_screening_exam/README"], "filenames": ["cnv/README.md", "ewas/README.md", "gatk_data_preprocessing/README.md", "gatk_germline_snp_indel/README.md", "gatk_mitochondria_snp_indel/README.md", "gatk_somatic_snv_indel/README.md", "index.rst", "mediation_analysis/README.md", "meta_analysis_funnel_plot/README.md", "mqtl/README.md", "mutational_signatures/README.md", "sv/README.md", "tableone/README.md", "telomere_length/README.md", "year_one_screening_exam/README.md"], "titles": ["CNV calling pipeline", "Epigenome-wide association study", "Data pre-processing for variant discovery GATK4", "GATK4 germline short variant discovery (SNPs + Indels)", "GATK4 mitochondria SNPs and INDELs pipeline", "GATK4 call somatic snvs indels", "phd-pipeline", "Causal mediation analysis", "Generate funnel plot for meta-analysis", "Methylation quantitative trait loci analysis", "Mutational Signatures Pipelines", "Calling Structural Variations", "Create summary stats table using R tableone package", "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)", "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam &#64;usc"], "terms": {"thi": [0, 2, 4, 5, 7, 8, 9, 10, 11], "i": [0, 4, 5, 6, 7, 9, 10, 11], "copi": 0, "number": [0, 9, 13], "variat": [0, 5], "from": [0, 1, 3, 10, 11], "76": [0, 11], "match": [0, 2, 4, 5, 11], "tumor": [0, 2, 5, 11], "normal": [0, 2, 5, 7, 11], "whole": [0, 2, 5, 11], "genom": [0, 2, 5, 11], "sequenc": [0, 2, 4, 5, 11], "wg": [0, 2, 3, 4, 5, 11, 13], "bam": [0, 5, 11], "38": [0, 11], "childhood": [0, 2, 4, 5, 9, 11, 13], "acut": [0, 2, 4, 5, 11, 13], "lymphoblast": [0, 2, 4, 5, 11, 13], "leukemia": [0, 2, 4, 5, 11, 13], "case": [0, 1, 2, 4, 5, 11, 13], "file": [0, 2, 10, 13], "have": [0, 5, 10], "been": [0, 10], "data": [0, 4, 5, 6], "check": [0, 2, 3, 4, 5], "out": [0, 2, 3, 4, 5], "best": [0, 3, 4], "practic": [0, 2, 3, 4], "document": [0, 2, 3, 5, 13], "cnv_somatic_pair_workflow": 0, "wdl": [0, 2, 3, 4, 5], "how": [0, 5], "part": 0, "sensit": 0, "detect": [0, 5, 10], "ratio": 0, "alter": [0, 5], "allel": [0, 5], "segment": 0, "ii": [0, 7], "legaci": 0, "tutori": [0, 2], "0": [0, 5, 9], "extractintervalfrompon": 0, "sh": [0, 2, 4, 5, 11], "1": [0, 3, 9, 11, 13], "9": [0, 4], "funcotateseg": 0, "workflow": [0, 1, 3, 4, 5, 6, 7, 9], "A": [0, 1], "figur": 0, "gitc": 0, "2": [0, 9, 13], "cnvkitaccess": 0, "3": [0, 11, 13], "iamp21cnvkitplot": 0, "cnvkittheta": 0, "cnvkitimporttheta": 0, "4": [0, 11, 13], "gisticcnvkitexportseg": 0, "gisticmergeseg": 0, "r": [0, 1, 4, 7, 9, 10], "run": [1, 2, 3, 13], "ewa": 1, "analys": [1, 7], "identifi": [1, 9], "all": [1, 7, 9, 13], "differenti": 1, "methyl": [1, 6, 7], "cpg": [1, 9], "posit": 1, "autosom": 1, "chromosom": 1, "we": [1, 5, 7, 9, 13], "fit": [1, 7, 9], "logist": 1, "regress": [1, 9], "model": [1, 5, 7, 9], "predict": [1, 9], "control": 1, "statu": [1, 7], "function": [1, 7, 9], "dna": [1, 9], "each": [1, 5, 9, 13], "remain": 1, "adjust": [1, 9], "sex": [1, 7], "batch": [1, 7], "effect": [1, 5, 7, 9], "cell": [1, 7], "type": [1, 7], "heterogen": [1, 7], "us": [1, 5, 7, 9, 10, 11, 13], "first": [1, 7, 9], "10": [1, 8], "princip": 1, "compon": 1, "deriv": 1, "refactor": 1, "58": 1, "genet": 1, "ancestri": [1, 7], "epistructur": 1, "fix": [1, 7, 9], "meta": [1, 6, 7, 9], "analysi": [1, 6], "were": [1, 2, 5, 7, 9, 13], "gener": [1, 13], "summari": [1, 6, 11], "estim": [1, 5, 6, 7], "result": [1, 7, 9, 13], "overlap": 1, "hree": 1, "differ": [1, 5, 10], "set": [1, 5], "packag": [1, 7, 9], "metafor": [1, 9], "public": [1, 7, 9], "link": [1, 7, 9], "script": [1, 5, 7, 9], "bidirect": 1, "manhattan": 1, "plot": [1, 6], "qq": 1, "sampl": [1, 5, 13], "cmplot": 1, "rmd": [1, 2, 10], "an": [2, 3, 4, 5, 7, 9, 11, 13], "hpc": [2, 3, 4, 5, 11], "50": [2, 5], "25": [2, 4, 5, 13], "illumina": [2, 13], "novaseq": [2, 13], "6000": [2, 13], "gatk": [2, 3, 5, 10, 13], "doc": [2, 4, 5], "more": [2, 3, 4, 5], "detail": [2, 3, 4, 5], "wa": [2, 5, 7, 9, 13], "written": [2, 5], "accord": [2, 3, 5], "follow": [2, 5], "offici": [2, 5], "version": 2, "http": [2, 3, 5, 8, 13], "github": [2, 3, 5, 13], "com": [2, 3, 5, 8, 13], "pipelin": [2, 3, 5, 11, 13], "blob": [2, 3], "master": [2, 3, 5], "task": [2, 3], "unmappedbamtoalignedbam": 2, "wholegenomegermlinesinglesampl": 2, "simpl": 2, "broadgsa": 2, "doc_arch": 2, "method": 2, "reference_implement": 2, "_pairedendsinglesamplewf_pipelin": 2, "md": 2, "sortandfixtag": 2, "other": [2, 5], "introduct": 2, "biowulf": 2, "nih": 2, "howto": 2, "bqsr": 2, "fastqc": 2, "pairedfastqtounmappedbam": 2, "validatesamfil": 2, "sortsam": 2, "bwa": 2, "sortalignedbam": 2, "validatealignedbam": 2, "checkifaltawarealign": 2, "mergebamalign": 2, "validatemergedbam": 2, "countread": 2, "validatesortedmarkedbam": 2, "baserecalibr": 2, "bqsrplot": 2, "haplotypecal": 3, "gvcf": 3, "mode": [3, 5], "broad": 3, "prod": 3, "jointgenotypingwf": 3, "broadinstitut": [3, 5], "warp": 3, "jointgenotypingtask": 3, "dna_seq": 3, "joint_genotyp": 3, "jointgenotyp": 3, "mitochondri": [4, 6], "larg": 4, "base": [4, 7], "subsetbamtochrm": 4, "revertsam": 4, "aligntomt": 4, "aligntoshiftedmt": 4, "collectwgsmetr": 4, "runcollectwgsmetricsr": 4, "callmt": 4, "germlin": [4, 5, 6], "5": [4, 10, 11], "callshiftedmt": 4, "6": [4, 5], "liftoverandcombinevcf": 4, "7": [4, 13], "mergestat": [4, 5], "8": 4, "initialfilt": 4, "splitmultiallelicsandremovenonpasssit": 4, "getcontamin": 4, "filtercontamin": 4, "mutat": [5, 6], "includ": [5, 7, 13], "singl": 5, "nucleotid": 5, "sna": 5, "insert": 5, "delet": 5, "preprocess": 5, "tree": 5, "mutect2_wdl": 5, "explan": 5, "deprec": 5, "new": 5, "pdf": 5, "featur": 5, "improv": 5, "what": 5, "tradit": 5, "entail": 5, "faq": 5, "onli": 5, "versu": 5, "doe": 5, "do": 5, "In": 5, "popul": 5, "frequenc": 5, "resourc": 5, "prior": 5, "read": [5, 13], "fraction": 5, "\u00bd": 5, "ar": 5, "suggest": 5, "het": 5, "evid": 5, "miss": 5, "": 5, "same": [5, 7], "addition": 5, "power": 5, "avail": 5, "exclud": 5, "rare": 5, "captur": 5, "individu": [5, 13], "specif": 5, "toward": 5, "being": 5, "The": [5, 7, 9], "simplifi": 5, "site": [5, 9], "gnomad": 5, "retain": 5, "ftp": 5, "gsapubftp": 5, "anonym": 5, "org": 5, "bundl": 5, "panel": 5, "pon": 5, "ha": 5, "vital": 5, "role": 5, "fill": 5, "gap": 5, "between": [5, 9], "catch": 5, "addit": [5, 9], "nois": 5, "like": 5, "map": 5, "somewhat": 5, "random": 5, "systemat": 5, "process": [5, 6, 10, 13], "interest": 5, "biostar": 5, "post": 5, "heterozygot": 5, "And": 5, "manifest": 5, "project": [5, 6], "creat": [5, 6], "contain": 5, "artifactu": 5, "tool": 5, "take": [5, 13], "multipl": [5, 9], "callset": 5, "produc": 5, "collat": 5, "present": 5, "two": [5, 7, 13], "default": 5, "min": 5, "count": 5, "argument": 5, "vcf": [5, 10], "common": 5, "level": [5, 7], "max": 5, "probabl": 5, "threshold": [5, 9, 13], "possibl": 5, "By": 5, "so": 5, "event": 5, "usual": 5, "correct": [5, 9], "behavior": 5, "handl": 5, "probabilist": 5, "via": 5, "human": 5, "much": 5, "refin": 5, "than": 5, "ani": 5, "could": 5, "final": [5, 13], "tabl": [5, 6, 13], "tell": 5, "u": 5, "rate": 5, "cross": 5, "If": 5, "15": 5, "error": 5, "19": 5, "mean": 5, "one": [5, 6], "everi": 5, "hundr": 5, "come": 5, "someon": 5, "els": 5, "percentag": 5, "our": 5, "floor": 5, "low": 5, "potenti": 5, "below": 5, "af": 5, "would": 5, "zero": 5, "judg": 5, "whether": 5, "thei": 5, "real": 5, "through": 5, "feed": 5, "can": 5, "properli": 5, "account": 5, "scatterinterv": 5, "mutect2scattergather0": 5, "mergevcf": 5, "mergebamout": 5, "orientationbia": 5, "createbwaimag": 5, "filteralignmentartifact": 5, "funcot": 5, "bcftoolfilt": 5, "some": 6, "wrote": 6, "my": 6, "research": 6, "cnv": 6, "epigenom": 6, "wide": 6, "associ": [6, 9], "studi": [6, 7, 8, 9], "pre": [6, 13], "variant": [6, 13], "discoveri": [6, 13], "gatk4": 6, "short": 6, "snp": [6, 9], "indel": 6, "snv": 6, "call": 6, "somat": 6, "causal": 6, "mediat": 6, "funnel": 6, "quantit": 6, "trait": 6, "loci": 6, "signatur": 6, "sv": 6, "stat": 6, "telomer": 6, "length": 6, "year": 6, "screen": 6, "exam": 6, "perform": 7, "signific": 7, "mqtl": [7, 9], "dmp": [7, 9], "pair": 7, "specifi": 7, "statist": 7, "distribut": 7, "after": 7, "condit": 7, "genotyp": [7, 9], "covari": 7, "outcom": 7, "given": 7, "separ": [7, 9], "paramet": 7, "main": 7, "input": 7, "which": 7, "comput": 7, "acm": 7, "ad": 7, "total": 7, "varianc": 7, "simul": 7, "quasi": 7, "bayesian": 7, "mont": 7, "carlo": 7, "approxim": 7, "conduct": [7, 9], "1000": 7, "time": 7, "altern": 7, "approach": 7, "non": 7, "parametr": 7, "bootstrap": 7, "also": 7, "appli": [7, 13], "valid": 7, "three": [7, 9, 11], "dataset": [7, 9], "summar": 7, "across": [7, 9], "path": 7, "diagram": 7, "show": 7, "confound": 7, "fig2": 8, "fig3": 8, "www": 8, "liebertpub": 8, "doi": 8, "ab": 8, "1089": 8, "jicm": 8, "2022": 8, "0586": 8, "depend": 9, "risk": 9, "matrix": 9, "eqtl": 9, "linear": 9, "code": [9, 14], "test": 9, "stringent": 9, "bonferroni": 9, "025": 9, "fdr": 9, "05": 9, "subsequ": 9, "analyz": 9, "mutect2": 10, "algorithm": 10, "step1": 10, "py": 10, "step2": 10, "step3": 10, "step4": 10, "signalsummari": 10, "signalkataegi": 10, "step5": 10, "softwar": [11, 13], "premergesvpersamplesmoov": 11, "smooveunionsit": 11, "smoovegenotyp": 11, "smoovepasteannot": 11, "premergesvpersamplemantaconfigur": 11, "mantaworkflow": 11, "mantaclean": 11, "dellypersamplecal": 11, "dellyprefilt": 11, "dellygeno": 11, "dellypostfilt": 11, "dellybcf2vcf": 11, "survivor": 11, "anno": 11, "exampl": 11, "graph": 11, "compar": 13, "paper": 13, "content": 13, "quantifi": 13, "ten": 13, "repeat": 13, "tcaggg": 13, "tgaggg": 13, "ttgggg": 13, "ttcggg": 13, "tttggg": 13, "ataggg": 13, "catggg": 13, "ctaggg": 13, "gtaggg": 13, "taaggg": 13, "tl": 13, "wsa": 13, "kb": 13, "ttaggg": 13, "ccctaa": 13, "consid": 13, "step": 13, "aggreg": 13, "output": 13, "clean": 13, "calcul": 13, "need": 13, "weight": 13, "averag": 13, "group": 13, "within": 13, "zd1": 13, "issu": 13, "had": 13, "high": 13}, "objects": {}, "objtypes": {}, "objnames": {}, "titleterms": {"cnv": 0, "call": [0, 4, 5, 11], "pipelin": [0, 4, 6, 10], "gatk": [0, 4], "somat": [0, 5], "cnvkit": 0, "preprocess": [0, 2], "creat": [0, 11, 12], "theta": 0, "input": 0, "gistic": 0, "epigenom": 1, "wide": 1, "associ": 1, "studi": 1, "data": [2, 13], "pre": 2, "process": 2, "variant": [2, 3, 5, 11], "discoveri": [2, 3, 5], "gatk4": [2, 3, 4, 5], "The": 2, "main": 2, "step": [2, 4, 5], "workflow": [2, 13], "A": [2, 5, 13], "figur": [2, 5], "from": [2, 4, 5, 13], "gitc": [2, 5], "0": [2, 4], "qc": 2, "1": [2, 4, 5], "fastq": 2, "unmap": [2, 4], "bam": [2, 4, 13], "2": [2, 4, 5, 11], "map": 2, "3": [2, 4, 5], "merg": 2, "4": [2, 4, 5], "markdupl": 2, "5": [2, 5], "base": 2, "qualiti": 2, "score": 2, "recalibr": 2, "germlin": 3, "short": [3, 5], "snp": [3, 4], "indel": [3, 4, 5], "mitochondria": 4, "download": 4, "": 4, "public": 4, "refer": 4, "databas": 4, "subset": 4, "whole": [4, 13], "genom": [4, 13], "just": 4, "read": 4, "remov": 4, "align": [4, 5], "inform": 4, "mark": 4, "duplic": 4, "collect": 4, "coverag": 4, "normal": [4, 13], "tumor": [4, 13], "estim": [4, 13, 14], "level": 4, "contamin": [4, 5], "snv": 5, "best": 5, "practic": 5, "scatter": 5, "interv": 5, "mutect2": 5, "parallel": 5, "gather": 5, "result": 5, "calcul": 5, "learn": 5, "orient": 5, "bia": 5, "filter": 5, "artifact": 5, "phd": [6, 14], "causal": 7, "mediat": 7, "analysi": [7, 8, 9], "gener": 8, "funnel": 8, "plot": [8, 13], "meta": 8, "methyl": 9, "quantit": 9, "trait": 9, "loci": 9, "mutat": 10, "signatur": 10, "sigprofil": 10, "pmsignatur": 10, "deconstructsig": 10, "signal": 10, "websit": 10, "fitm": 10, "structur": 11, "variat": 11, "smoov": 11, "lumpi": 11, "manta": 11, "delli": 11, "find": 11, "were": 11, "least": 11, "caller": 11, "annot": 11, "venn": 11, "diagram": [11, 14], "histogram": 11, "html": 11, "file": 11, "summari": 12, "stat": 12, "tabl": 12, "us": 12, "r": 12, "tableon": 12, "packag": 12, "telomer": 13, "length": 13, "match": 13, "sequenc": 13, "background": 13, "telomerehunt": 13, "telseq": 13, "exampl": 13, "correl": 13, "sa": 14, "script": 14, "spider": 14, "i": 14, "prepar": 14, "epidemiologi": 14, "screen": 14, "exam": 14, "usc": 14, "appli": 14, "regress": 14, "predict": 14, "theori": 14}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinx.ext.viewcode": 1, "sphinx": 57}, "alltitles": {"Mutational Signatures Pipelines": [[10, "mutational-signatures-pipelines"]], "Sigprofiler": [[10, "sigprofiler"]], "pmsignature": [[10, "pmsignature"]], "deconstructSigs": [[10, "deconstructsigs"]], "Signal website": [[10, "signal-website"]], "FitMS": [[10, "fitms"]], "Methylation quantitative trait loci analysis": [[9, "methylation-quantitative-trait-loci-analysis"]], "Generate funnel plot for meta-analysis": [[8, "generate-funnel-plot-for-meta-analysis"]], "Causal mediation analysis": [[7, "causal-mediation-analysis"]], "Calling Structural Variations": [[11, "calling-structural-variations"]], "Smoove (lumpy)": [[11, "smoove-lumpy"]], "Manta": [[11, "manta"]], "Delly": [[11, "delly"]], "Finding variants that were called by at least 2 callers": [[11, "finding-variants-that-were-called-by-at-least-2-callers"]], "Annotate variants": [[11, "annotate-variants"]], "Create Venn diagram, histograms and html files": [[11, "create-venn-diagram-histograms-and-html-files"]], "Create summary stats table using R tableone package": [[12, "create-summary-stats-table-using-r-tableone-package"]], "A workflow to estimate telomere length from matched tumor normal whole genome sequencing data (BAMs)": [[13, "a-workflow-to-estimate-telomere-length-from-matched-tumor-normal-whole-genome-sequencing-data-bams"]], "Background": [[13, "background"]], "TelomereHunter": [[13, "telomerehunter"]], "TelSeq": [[13, "telseq"]], "Workflow": [[13, "workflow"]], "Example correlation plot": [[13, "example-correlation-plot"]], "SAS scripts and spider diagram that I prepared for the epidemiology phd screening exam @usc": [[14, "sas-scripts-and-spider-diagram-that-i-prepared-for-the-epidemiology-phd-screening-exam-usc"]], "apply exam - regression estimation and prediction": [[14, "apply-exam-regression-estimation-and-prediction"]], "theory - epidemiology": [[14, "theory-epidemiology"]], "CNV calling pipeline": [[0, "cnv-calling-pipeline"]], "GATK somatic CNV calling pipeline:": [[0, "gatk-somatic-cnv-calling-pipeline"]], "CNVkit pipeline": [[0, "cnvkit-pipeline"]], "Preprocessing to create Theta inputs": [[0, "preprocessing-to-create-theta-inputs"]], "Gistic pipeline": [[0, "gistic-pipeline"]], "Data pre-processing for variant discovery GATK4": [[2, "data-pre-processing-for-variant-discovery-gatk4"]], "The main steps in the preprocessing workflow. A figure from GITC.": [[2, "the-main-steps-in-the-preprocessing-workflow-a-figure-from-gitc"]], "Step 0: QC": [[2, "step-0-qc"]], "Step 1: Fastqs to Unmapped BAMs": [[2, "step-1-fastqs-to-unmapped-bams"]], "Step 2: Unmapped BAMs to Mapped BAMs": [[2, "step-2-unmapped-bams-to-mapped-bams"]], "Step 3: Merge Unmapped BAMs and Mapped BAMs": [[2, "step-3-merge-unmapped-bams-and-mapped-bams"]], "Step 4: MarkDuplicates": [[2, "step-4-markduplicates"]], "Step 5: Base Quality Score Recalibration": [[2, "step-5-base-quality-score-recalibration"]], "GATK4 call somatic snvs indels": [[5, "gatk4-call-somatic-snvs-indels"]], "Best practices for somatic short variant discovery. A figure from GITC.": [[5, "best-practices-for-somatic-short-variant-discovery-a-figure-from-gitc"]], "Step 1: Scatter interval for Mutect2 parallel calling": [[5, "step-1-scatter-interval-for-mutect2-parallel-calling"]], "Step 2: Gather results from scattered intervals": [[5, "step-2-gather-results-from-scattered-intervals"]], "Step 3: Calculate contamination and learn orientation bias": [[5, "step-3-calculate-contamination-and-learn-orientation-bias"]], "Step 4: Filter alignment artifacts": [[5, "step-4-filter-alignment-artifacts"]], "Step 5: Filter variants": [[5, "step-5-filter-variants"]], "phd-pipeline": [[6, "phd-pipeline"]], "GATK4 mitochondria SNPs and INDELs pipeline": [[4, "gatk4-mitochondria-snps-and-indels-pipeline"]], "Step 0: Download GATK\u2019s public reference databases": [[4, "step-0-download-gatk-s-public-reference-databases"]], "Step 1: Subset a whole genome bam to just Mitochondria reads and remove alignment information": [[4, "step-1-subset-a-whole-genome-bam-to-just-mitochondria-reads-and-remove-alignment-information"]], "Step 2: Align unmapped bam, mark duplicates, collect coverage information, and call SNPs and INDELs from normal bams": [[4, "step-2-align-unmapped-bam-mark-duplicates-collect-coverage-information-and-call-snps-and-indels-from-normal-bams"]], "Step 3: Call SNPs and INDELs from tumor bams": [[4, "step-3-call-snps-and-indels-from-tumor-bams"]], "Step 4: Estimate levels of contamination in mitochondria": [[4, "step-4-estimate-levels-of-contamination-in-mitochondria"]], "Epigenome-wide association study": [[1, "epigenome-wide-association-study"]], "GATK4 germline short variant discovery (SNPs + Indels)": [[3, "gatk4-germline-short-variant-discovery-snps-indels"]]}, "indexentries": {}})