# Epigenome-wide association study

A workflow that runs EWAS analyses to identify ALL-associated differentially methylated CpG positions on autosomal chromosomes. We fitted a logistic regression model predicting ALL case/control status as a function of DNA methylation at each remaining CpG, adjusting for sex, batch effect, cell type heterogeneity using the first 10 principal components derived from ReFACTor (58) and genetic ancestry using the first 10 principal components derived from EPISTRUCTURE. Fixed-effect meta-analysis models were used to generate summary effect estimates for the EWAS results of the CpGs overlapping hree different study sets—using the R package ‘metafor’.

- Study publication: [link](https://academic.oup.com/hmg/article/31/21/3741/6611023#378311310)
- EWAS script: [link](./ewas.r)
- Scripts for Bidirectional Manhattan plot and QQ plot [link](./CMplot.Rmd)
- Sample plots generated by `CMplot.Rmd`:

<p align="center">
<img src='../_static/ewas_Fig.2.png' width='500'>
</p>
