<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

    <title>Data pre-processing for variant discovery GATK4 &#8212; phd-pipeline 1.0.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/bootstrap-sphinx.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.1e8bd061cd6da7fc9cf755528e8ffc24.min.css" />
    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/sphinx_highlight.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/design-tabs.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
<meta charset='utf-8'>
<meta http-equiv='X-UA-Compatible' content='IE=edge,chrome=1'>
<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1'>
<meta name="apple-mobile-web-app-capable" content="yes">
<script type="text/javascript" src="../_static/js/jquery-1.12.4.min.js"></script>
<script type="text/javascript" src="../_static/js/jquery-fix.js"></script>
<script type="text/javascript" src="../_static/bootstrap-3.4.1/js/bootstrap.min.js"></script>
<script type="text/javascript" src="../_static/bootstrap-sphinx.js"></script>

  </head><body>

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="../index.html">
          phd-pipeline</a>
        <span class="navbar-text navbar-version pull-left"><b></b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="../index.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"></ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">Data pre-processing for variant discovery GATK4</a><ul>
<li><a class="reference internal" href="#the-main-steps-in-the-preprocessing-workflow-a-figure-from-gitc">The main steps in the preprocessing workflow. A figure from GITC.</a></li>
<li><a class="reference internal" href="#step-0-qc">Step 0: QC</a></li>
<li><a class="reference internal" href="#step-1-fastqs-to-unmapped-bams">Step 1: Fastqs to Unmapped BAMs</a></li>
<li><a class="reference internal" href="#step-2-unmapped-bams-to-mapped-bams">Step 2: Unmapped BAMs to Mapped BAMs</a></li>
<li><a class="reference internal" href="#step-3-merge-unmapped-bams-and-mapped-bams">Step 3: Merge Unmapped BAMs and Mapped BAMs</a></li>
<li><a class="reference internal" href="#step-4-markduplicates">Step 4: MarkDuplicates</a></li>
<li><a class="reference internal" href="#step-5-base-quality-score-recalibration">Step 5: Base Quality Score Recalibration</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="../_sources/gatk_data_preprocessing/README.md.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="../search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">
    <div class="body col-md-12 content" role="main">
      
  <section class="tex2jax_ignore mathjax_ignore" id="data-pre-processing-for-variant-discovery-gatk4">
<h1>Data pre-processing for variant discovery GATK4<a class="headerlink" href="#data-pre-processing-for-variant-discovery-gatk4" title="Permalink to this heading">¶</a></h1>
<p>An HPC workflow that pre-processes 50 matched tumor-normal whole genome sequencing (WGS) fastq files from 25 childhood acute lymphoblastic leukemia cases. WGS data were from Illumina NovaSeq 6000 Sequencing. Check out <a class="reference external" href="https://gatk.broadinstitute.org/hc/en-us/articles/360035535912-Data-pre-processing-for-variant-discovery">gatk doc</a> for more details.</p>
<p>This workflow was written according to the following GATK official WDLs:</p>
<ul class="simple">
<li><p>Detailed versions:</p>
<ul>
<li><p>https://github.com/gatk-workflows/gatk4-genome-processing-pipeline/blob/master/tasks/UnmappedBamToAlignedBam.wdl</p></li>
<li><p>https://github.com/gatk-workflows/gatk4-genome-processing-pipeline/blob/master/WholeGenomeGermlineSingleSample.wdl</p></li>
</ul>
</li>
<li><p>Simple version:</p>
<ul>
<li><p>https://github.com/gatk-workflows/gatk4-data-processing/blob/master/processing-for-variant-discovery-gatk4.wdl</p></li>
<li><p>Detailed documents for this simple version: https://github.com/broadgsa/gatk/blob/master/doc_archive/methods/Reference_implementation:_PairedEndSingleSampleWf_pipeline.md#SortAndFixTags</p></li>
</ul>
</li>
<li><p>Other tutorials:</p>
<ul>
<li><p><a class="reference external" href="https://hpc.nih.gov/training/gatk_tutorial/bqsr.html">A practical introduction to GATK 4 on Biowulf (NIH HPC)</a></p></li>
<li><p><a class="reference external" href="https://github.com/broadgsa/gatk/blob/master/doc_archive/tutorials/(howto)_Recalibrate_base_quality_scores_%3D_run_BQSR.md">(howto) Recalibrate base quality scores = run BQSR</a></p></li>
</ul>
</li>
</ul>
<section id="the-main-steps-in-the-preprocessing-workflow-a-figure-from-gitc">
<h2>The main steps in the preprocessing workflow. A figure from GITC.<a class="headerlink" href="#the-main-steps-in-the-preprocessing-workflow-a-figure-from-gitc" title="Permalink to this heading">¶</a></h2>
<p align="center">
<img src='../_static/MainStepsInPreprocessingWorkflow.png' width='250'>
</p>
</section>
<section id="step-0-qc">
<h2>Step 0: QC<a class="headerlink" href="#step-0-qc" title="Permalink to this heading">¶</a></h2>
<ul class="simple">
<li><p><a class="reference download internal" download="" href="../_downloads/720a4ffacbc4af2be61555162246ea29/0.1.Fastqc.sh"><span class="xref download myst">0.1.Fastqc.sh</span></a></p></li>
</ul>
</section>
<section id="step-1-fastqs-to-unmapped-bams">
<h2>Step 1: Fastqs to Unmapped BAMs<a class="headerlink" href="#step-1-fastqs-to-unmapped-bams" title="Permalink to this heading">¶</a></h2>
<ul class="simple">
<li><p><a class="reference download internal" download="" href="../_downloads/b46aea83dd1701044fe60456e6a84c1c/1.1.PairedFastqToUnmappedBam.sh"><span class="xref download myst">1.1.PairedFastqToUnmappedBam.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/92ca7c195fc347928e0feadfc166b60a/1.2.ValidateSamFile.sh"><span class="xref download myst">1.2.ValidateSamFile.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/f8035ac3e59971270fdeb555c306bdf4/1.3.SortSam.sh"><span class="xref download myst">1.3.SortSam.sh</span></a></p></li>
</ul>
</section>
<section id="step-2-unmapped-bams-to-mapped-bams">
<h2>Step 2: Unmapped BAMs to Mapped BAMs<a class="headerlink" href="#step-2-unmapped-bams-to-mapped-bams" title="Permalink to this heading">¶</a></h2>
<ul class="simple">
<li><p><a class="reference download internal" download="" href="../_downloads/ca4b005312338586688926cd214e88b9/2.1.Bwa.sh"><span class="xref download myst">2.1.Bwa.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/db7595df5856e4ca38e5e678cbe23a39/2.2.SortAlignedBam.sh"><span class="xref download myst">2.2.SortAlignedBam.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/356857112fc22b9af8219cf6ee04f15d/2.3.ValidateAlignedBam.sh"><span class="xref download myst">2.3.ValidateAlignedBam.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/829d405024b8f014d0b3dea3f7299b07/2.4.CheckIfAltAwareAlign.sh"><span class="xref download myst">2.4.CheckIfAltAwareAlign.sh</span></a></p></li>
</ul>
</section>
<section id="step-3-merge-unmapped-bams-and-mapped-bams">
<h2>Step 3: Merge Unmapped BAMs and Mapped BAMs<a class="headerlink" href="#step-3-merge-unmapped-bams-and-mapped-bams" title="Permalink to this heading">¶</a></h2>
<ul class="simple">
<li><p><a class="reference download internal" download="" href="../_downloads/f73ffcf1d78b54b1745001032044c815/3.1.MergeBamAlignment.sh"><span class="xref download myst">3.1.MergeBamAlignment.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/4e09d86921c1f1ec848b555dcb4a850a/3.2.ValidateMergedBam.sh"><span class="xref download myst">3.2.ValidateMergedBam.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/68b8e598994f2cd5fca702866ebc1d92/3.3.CountReads.sh"><span class="xref download myst">3.3.CountReads.sh</span></a></p></li>
</ul>
</section>
<section id="step-4-markduplicates">
<h2>Step 4: MarkDuplicates<a class="headerlink" href="#step-4-markduplicates" title="Permalink to this heading">¶</a></h2>
<ul class="simple">
<li><p><a class="reference download internal" download="" href="../_downloads/63f90a062dcd0b9730cb179b75cb18a4/4.1.MarkDuplicates.sh"><span class="xref download myst">4.1.MarkDuplicates.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/e6a9ae9c2cc189a5cbcfaab8aa19feac/4.2.SortAndFixTags.sh"><span class="xref download myst">4.2.SortAndFixTags.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/0a20d787265f8bc1be09e4249abac28a/4.3.ValidateSortedMarkedBam.sh"><span class="xref download myst">4.3.ValidateSortedMarkedBam.sh</span></a></p></li>
</ul>
</section>
<section id="step-5-base-quality-score-recalibration">
<h2>Step 5: Base Quality Score Recalibration<a class="headerlink" href="#step-5-base-quality-score-recalibration" title="Permalink to this heading">¶</a></h2>
<ul class="simple">
<li><p><a class="reference download internal" download="" href="../_downloads/7a4739434faf6510d21e3116c7795e1e/5.1.BaseRecalibrator.sh"><span class="xref download myst">5.1.BaseRecalibrator.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/d5a3dc1bc5dd87382595185cd5ac5b90/5.2.BqsrPlot.sh"><span class="xref download myst">5.2.BqsrPlot.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/17f7a1e45137237d1138b4e904c0bb3e/5.3.BqsrPlot.Rmd"><span class="xref download myst">5.3.BqsrPlot.Rmd</span></a></p></li>
</ul>
</section>
</section>


    </div>
      
  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2023, Keren Xu.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 5.3.0.<br/>
    </p>
  </div>
</footer>
  </body>
</html>