

<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

    <title>GATK4 mitochondria SNPs and INDELs pipeline &#8212; phd-pipeline 1.0.0 documentation</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=e353d410970836974a52" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=e353d410970836974a52" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.1e8bd061cd6da7fc9cf755528e8ffc24.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=e353d410970836974a52" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52" />

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/sphinx_highlight.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script src="../_static/design-tabs.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'gatk_mitochondria_snp_indel/README';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search..."
         aria-label="Search..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="../index.html">
  
  
  
  
  
    <p class="title logo__title">phd-pipeline 1.0.0 documentation</p>
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">


<a href="https://github.com/XUKEREN/phd-pipeline" target="_blank"
   class="btn btn-sm btn-source-repository-button"
   title="Source repository"
   data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>

</a>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/gatk_mitochondria_snp_indel/README.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>GATK4 mitochondria SNPs and INDELs pipeline</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-0-download-gatk-s-public-reference-databases">Step 0: Download GATK’s public reference databases</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-1-subset-a-whole-genome-bam-to-just-mitochondria-reads-and-remove-alignment-information">Step 1: Subset a whole genome bam to just Mitochondria reads and remove alignment information</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-2-align-unmapped-bam-mark-duplicates-collect-coverage-information-and-call-snps-and-indels-from-normal-bams">Step 2: Align unmapped bam, mark duplicates, collect coverage information, and call SNPs and INDELs from normal bams</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-3-call-snps-and-indels-from-tumor-bams">Step 3: Call SNPs and INDELs from tumor bams</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-4-estimate-levels-of-contamination-in-mitochondria">Step 4: Estimate levels of contamination in mitochondria</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="gatk4-mitochondria-snps-and-indels-pipeline">
<h1>GATK4 mitochondria SNPs and INDELs pipeline<a class="headerlink" href="#gatk4-mitochondria-snps-and-indels-pipeline" title="Permalink to this heading">#</a></h1>
<p>This is an HPC workflow to call SNPs and INDELs in the mitochondrial genome from matched tumor-normal whole genome sequencing (WGS) data from 25 childhood acute lymphoblastic leukemia cases. This workflow is largely based on <a class="reference external" href="https://github.com/broadinstitute/gatk/blob/master/scripts/mitochondria_m2_wdl/MitochondriaPipeline.wdl">GATK’s best practice WDL</a>. Check out <a class="reference external" href="https://github.com/broadinstitute/gatk-docs/blob/master/blog-2012-to-2019/2019-03-05-New!_Mitochondrial_Analysis_with_Mutect2.md?id=23598">GATK’s doc</a> for more details.</p>
<section id="step-0-download-gatk-s-public-reference-databases">
<h2>Step 0: Download GATK’s public reference databases<a class="headerlink" href="#step-0-download-gatk-s-public-reference-databases" title="Permalink to this heading">#</a></h2>
<ul class="simple">
<li><p><a class="reference download internal" download="" href="../_downloads/3b9c8c897b77d932c967ec2c1606d9ef/0.1.Download.sh"><span class="xref download myst">0.1.Download.sh</span></a></p></li>
</ul>
</section>
<section id="step-1-subset-a-whole-genome-bam-to-just-mitochondria-reads-and-remove-alignment-information">
<h2>Step 1: Subset a whole genome bam to just Mitochondria reads and remove alignment information<a class="headerlink" href="#step-1-subset-a-whole-genome-bam-to-just-mitochondria-reads-and-remove-alignment-information" title="Permalink to this heading">#</a></h2>
<ul class="simple">
<li><p><a class="reference download internal" download="" href="../_downloads/7f7df36bb1fe9eed267f972836655c78/1.1.SubsetBamToChrM.sh"><span class="xref download myst">1.1.SubsetBamToChrM.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/3bdaaee900ecfb6a183719730c6c312a/1.2.RevertSam.sh"><span class="xref download myst">1.2.RevertSam.sh</span></a></p></li>
</ul>
</section>
<section id="step-2-align-unmapped-bam-mark-duplicates-collect-coverage-information-and-call-snps-and-indels-from-normal-bams">
<h2>Step 2: Align unmapped bam, mark duplicates, collect coverage information, and call SNPs and INDELs from normal bams<a class="headerlink" href="#step-2-align-unmapped-bam-mark-duplicates-collect-coverage-information-and-call-snps-and-indels-from-normal-bams" title="Permalink to this heading">#</a></h2>
<ul class="simple">
<li><p><a class="reference download internal" download="" href="../_downloads/29c526c3fbb59075c12fdb29bd184c9c/2.1.AlignToMt.sh"><span class="xref download myst">2.1.AlignToMt.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/fc7cddd2f0f4b8f55cc054e6cd0738c8/2.2.AlignToShiftedMt.sh"><span class="xref download myst">2.2.AlignToShiftedMt.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/c5bd9a0de21980299e64101b273b7288/2.3.1.CollectWgsMetrics.sh"><span class="xref download myst">2.3.1.CollectWgsMetrics.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/a12578c54b06a207c16f53f3bf547829/2.3.2.CollectWgsMetrics.R"><span class="xref download myst">2.3.2.CollectWgsMetrics.R</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/90b5e9aafcefcb5e7a0a41437797ac2a/2.3.3.RunCollectWgsMetricsR.sh"><span class="xref download myst">2.3.3.RunCollectWgsMetricsR.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/a03184b8ecc10dafed8f80595ab12cd1/2.4.CallMt.Germline.sh"><span class="xref download myst">2.4.CallMt.Germline.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/dc79b89f65b7c1b5bdc86d691c98cfdc/2.5.CallShiftedMt.Germline.sh"><span class="xref download myst">2.5.CallShiftedMt.Germline.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/0fff000c79100188ea2dc323e616570d/2.6.LiftoverAndCombineVcfs.sh"><span class="xref download myst">2.6.LiftoverAndCombineVcfs.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/c2f090fcfca6276140ddfbee84f001b5/2.7.MergeStats.sh"><span class="xref download myst">2.7.MergeStats.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/c81804a4b1f322d5ad41ea646d098149/2.8.InitialFilter.sh"><span class="xref download myst">2.8.InitialFilter.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/987523ca8780c3f0f4c6c099229d04b4/2.9.SplitMultiAllelicsAndRemoveNonPassSites.sh"><span class="xref download myst">2.9.SplitMultiAllelicsAndRemoveNonPassSites.sh</span></a></p></li>
</ul>
</section>
<section id="step-3-call-snps-and-indels-from-tumor-bams">
<h2>Step 3: Call SNPs and INDELs from tumor bams<a class="headerlink" href="#step-3-call-snps-and-indels-from-tumor-bams" title="Permalink to this heading">#</a></h2>
<ul class="simple">
<li><p><a class="reference download internal" download="" href="../_downloads/a4fdbcd9b8595dbc5d72ba8b8ca40df5/3.1.CallMt.Tumor.sh"><span class="xref download myst">3.1.CallMt.Tumor.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/ff86fbd924899011a880b5d277bc3569/3.2.CallShiftedMt.Tumor.sh"><span class="xref download myst">3.2.CallShiftedMt.Tumor.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/5161dfff3cd83a16cdece9c6b05a5ae4/3.3.LiftoverAndCombineVcfs.Tumor.sh"><span class="xref download myst">3.3.LiftoverAndCombineVcfs.Tumor.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/e467cda7951e09dcafc01e27a1932523/3.4.MergeStats.Tumor.sh"><span class="xref download myst">3.4.MergeStats.Tumor.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/d0748268850d878b81184a532657e778/3.5.InitialFilter.Tumor.sh"><span class="xref download myst">3.5.InitialFilter.Tumor.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/7f727a82374ec25f43d668cb00cf407f/3.6.SplitMultiAllelicsAndRemoveNonPassSites.sh"><span class="xref download myst">3.6.SplitMultiAllelicsAndRemoveNonPassSites.sh</span></a></p></li>
</ul>
</section>
<section id="step-4-estimate-levels-of-contamination-in-mitochondria">
<h2>Step 4: Estimate levels of contamination in mitochondria<a class="headerlink" href="#step-4-estimate-levels-of-contamination-in-mitochondria" title="Permalink to this heading">#</a></h2>
<ul class="simple">
<li><p><a class="reference download internal" download="" href="../_downloads/55089db83712f6d1e76d4a1a18c0daa8/4.1.GetContamination.sh"><span class="xref download myst">4.1.GetContamination.sh</span></a></p></li>
<li><p><a class="reference download internal" download="" href="../_downloads/8534b7c0ffb0e3c13f53bc2f880080bb/4.2.FilterContamination.sh"><span class="xref download myst">4.2.FilterContamination.sh</span></a></p></li>
</ul>
</section>
</section>


                </article>
              

              
              
                <footer class="bd-footer-article">
                  
<div class="footer-article-items footer-article__inner">
  
    <div class="footer-article-item"><!-- Previous / next buttons -->
<div class="prev-next-area">
</div></div>
  
</div>

                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-0-download-gatk-s-public-reference-databases">Step 0: Download GATK’s public reference databases</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-1-subset-a-whole-genome-bam-to-just-mitochondria-reads-and-remove-alignment-information">Step 1: Subset a whole genome bam to just Mitochondria reads and remove alignment information</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-2-align-unmapped-bam-mark-duplicates-collect-coverage-information-and-call-snps-and-indels-from-normal-bams">Step 2: Align unmapped bam, mark duplicates, collect coverage information, and call SNPs and INDELs from normal bams</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-3-call-snps-and-indels-from-tumor-bams">Step 3: Call SNPs and INDELs from tumor bams</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#step-4-estimate-levels-of-contamination-in-mitochondria">Step 4: Estimate levels of contamination in mitochondria</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Keren Xu
</p>

  </div>
  
  <div class="footer-item">
    
  <p class="copyright">
    
      © Copyright 2023, Keren Xu.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=e353d410970836974a52"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>